---
title: "Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation"
author: 
  - name: "Banabithi Bose"
    affiliation: "Northwestern University"
  - name: "Catherine Stanhope"
    affiliation: "Northwestern University"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Decoding Ancestry}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=8, fig.height=6)
```
## Introduction

The pursuit of ancestral origins using genetic data has become increasingly accessible and sophisticated with the advent of genomics technology. Our R package GXwasR enables the identification of individuals of divergent ancestry through genotype datasets. This vignette provides an in-depth look at how GXwasR facilitates ancestry estimation, particularly focusing on its `AncestryCheck`() function.

## Identifying Divergent Ancestry
The core of ancestry estimation lies in combining the genotypes of a study population with those from a reference dataset, such as the HapMap or 1000 Genomes studies. This integration allows for the detection of population structure down to the level of the reference dataset, revealing continental-scale ancestry differences.

## Accessing GXwasR and Simulated Genotype Datasets
GXwasR is readily available on the Comprehensive R Archive Network (CRAN) and can be installed using install.packages("GXwasR"). The package includes simulated genotype dataset prefixed as **GXwasR_example** as plink bed files (i.e., .bed, .fam and .bim)[1], ideal for users to experiment with. This simulated dataset consists of genotypes from 276 individuals across 26,527 genetic markers, providing a realistic scenario for testing the functionality.

Let's see more details of this genotype dataset that we will call 'Study data'.
```{r}
source("/projects/b1137/BBose/GXwasR/R/GXwasR_main_functions.R")
source("/projects/b1137/BBose/GXwasR/R/GXwasR_helper_functions.R")
source("/projects/b1137/BBose/Panscan_codes/AncestryFunction2024_Edited._v2R.R")
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir <- tempdir()
finput <- "GXwasR_example"
x <- PlinkSummary(DataDir,ResultDir, finput)
```

## AncestryCheck Function Overview
  
`AncestryCheck` is a comprehensive R function designed for comprehensive ancestry analysis in genetic studies. By leveraging plink software via R platform,it seamlessly integrates several critical steps: SNP filtering for data integrity, linkage disequilibrium (LD) pruning to reduce data complexity, correction of chromosome mismatches, and checking for allele flips to ensure data accuracy. Additionally, it adeptly merges study and reference datasets, conducts Principal Component Analysis (PCA) to unravel population structures, and incorporates outlier detection and visualization. This function is pivotal in comparing study samples' self-reported ancestry against reference populations, enabling a detailed analysis of population structure and the identification of divergent ancestries.
  
## Implemented workflow of ancestry estimation in AncestryCheck
The `AncestryCheck`() function in the GXwasR package is a sophisticated tool designed for ancestry analysis in genetic studies. It provides a comprehensive approach, from data preparation to advanced statistical analysis, for estimating ancestral backgrounds. Let's delve into the technical aspects and workflow of this function:

### Data Preparation:
1. **Directories and initial quality check:** The '`AncestryCheck`' function requires specific directories for inputting PLINK binary files and for saving output files. At this stage, the input genotype dataset, i.e., the study dataset, should have already undergone thorough quality control. GXwasR offers a range of functions to facilitate this preliminary quality check (QC) of genotype datasets, essential for accurate ancestry estimation. This QC process typically includes filtering out multi-allelic variants, selecting SNPs based on genotype call rate, filtering for specific chromosomes, SNP filtering grounded in minor allele frequency, adherence to Hardy-Weinberg equilibrium, and sample selection based on criteria like call rate, low missing rate, and heterozygosity thresholds. It's important to apply these filters in the appropriate sequence for optimal dataset preparation. For a more detailed understanding of this QC pipeline, users can refer to the comprehensive guide available at `www.hgvjhvkjfbkjjrjttsjdkuffij`.

2. **Genomic Build and X-Chromosome Compatibility:** Users of the `AncestryCheck` function must ensure that both their study and reference datasets are aligned on the same genomic build, with consistent chromosome coding across the entire genome. This alignment is crucial, particularly since sex chromosomes (X and Y) often have different encoding conventions in genomic datasets. The `AncestryCheck` function is equipped to implicitly handle variations in sex chromosome encoding. However, users are advised to employ tools like Liftover or CrossMap for verifying and aligning genomic builds and chromosome codes between their study and reference datasets, thus ensuring compatibility and accuracy in the ancestry estimation process.

2. **Reference Dataset Selection:** The function offers users the flexibility to select from two established reference populations, such as, **HapMap phase III**[2] or the **1000 Genomes phase 3**[3] Projects. It is specifically designed to work seamlessly with the preprocessed genotype datasets of these reference panels. For users seeking to download and process these datasets, a comprehensive guidance was provided through a dedicated vignette. This detailed resource, which can be accessed at `www.gghdhgmhhgjgktddyufufu`, offers step-by-step instructions on how to effectively prepare these reference datasets for use in ancestry analysis, ensuring users can make the most out of their genetic studies with the selected reference populations.

### SNP Filtering: 
The function incorporates a crucial step of SNP filtering for both study and reference datasets. This filtering process is particularly focused on excluding SNPs that do not conform to A-T or G-C base pairings. By implementing this filtering, `AncestryCheck` ensures the overall quality and reliability of the genetic data from both study and reference datasets. This step is vital in maintaining the integrity of the analysis, as it helps in eliminating potentially problematic SNPs that could skew the ancestry estimation results.

### LD Pruning: 
The function includes a critical step of LD-based filtering for both the study and reference datasets, which is essential in simplifying the genetic data by removing variants in high linkage disequilibrium. This process is especially significant for conducting accurate PCA on the study dataset SNPs. The default setting targets variants in LD with an `r2` value greater than 0.2 within a 50kb window. However, `AncestryCheck` offers users the flexibility to adjust these parameters according to their specific research needs. Users can choose whether or not to apply LD-based filtering and can modify parameters like the `r2` threshold and the `window size`. This adaptability ensures that the function can be tailored to a variety of study designs and research objectives, making it a versatile tool for ancestry analysis.

The '`AncestryCheck` then utilizes the list of pruned variants from the study data to proportionately adjust the reference dataset, aligning it to the scale of the study data. This harmonization is crucial for accurate comparative analysis and ensures consistency in the ancestral estimation process.

### Filter for known high-LD structure.
The '`AncestryCheck`' provides an additional layer of refinement by offering the option to filter out genomic ranges characterized by high-LD [4]. This feature is particularly beneficial for genomic builds hg19 and hg38, as it targets and removes regions within these builds that are known for their high-LD structures. These specific high-LD region files are already incorporated into the GXwasR package, readily accessible for use in `AncestryCheck`. This targeted filtering is crucial in further enhancing the quality and accuracy of the genetic data being analyzed, ensuring that the resulting ancestry estimation is as precise and reliable as possible.

### Genotype Data Processing for Merge:
The function includes a meticulous process to prepare genotype data for merging, addressing key issues that could otherwise compromise the accuracy of the analysis:

**Correcting Mismatches:** The function systematically corrects mismatches in chromosome IDs, positions, and allele annotations. This step is essential to ensure that the study and reference data align accurately, providing a reliable basis for subsequent analysis. By rectifying these discrepancies, `AncestryCheck` safeguards against potential errors that could arise from inconsistencies in the datasets.

**Handling Allele Flips:** Allele flips are a frequent complication in genotype data, where alleles may be reported in reverse order. The function is adept at identifying and rectifying these flips, thereby preserving the integrity of the genetic information. Given the importance of accurately matching SNPs, the ability to correct allele flips is crucial for error-free dataset merging.

**Dealing with Sex Chromosomes:** The encoding of sex chromosomes often differs from autosomal chromosomes, which can introduce additional complexity in matching study and reference datasets. `AncestryCheck` is designed to handle these variations effectively.

**Removing Mismatches Post-Flipping:** After the allele flipping process, the function performs a thorough check to identify any remaining mismatches. Any alleles that are still not aligned correctly after the flipping process are identified and removed from the reference dataset. This final step ensures that only perfectly matched SNPs are retained for the merged dataset, thereby enhancing the accuracy of the ancestral analysis.

Through these comprehensive steps, `AncestryCheck` ensures that the genotype datasets are optimally prepared for merging, laying a solid foundation for accurate and reliable ancestry estimation.

### Principal Component Analysis (PCA):
**Merging Datasets:** After preprocessing, the study and reference datasets are merged.

**Executing PCA:** PCA is then performed on the combined dataset, providing insights into the genetic variance and ancestral components. 

By implementing PCA on the harmonized genetic data, `AncestryCheck` effectively translates the nuanced genetic variations into interpretable patterns, facilitating a deeper understanding of the ancestral backgrounds inherent in the study population. This step is fundamental in transforming raw genetic data into meaningful insights about human ancestry and genetic diversity.

### Reference Populations:

**HapMap:** 
We combined and used HapMap data for the following populations:
AFR = ASW (African Ancestry in SW USA), LWK (Luhya in Webuye, Kenya), MKK (Maasai in Kinyawa, Kenya), and YRI (Yoruba in Ibadan, Nigeria).
EAS = CHB (Han Chinese in Beijing, China), CHD (Chinese in Metropolitan Denver, CO, USA), and JPT (Japanese in Tokyo, Japan).
EUR = CEU (CEPH/Utah Collection) and TSI (Toscani in Italia).
SAS = GIH (Gujarati Indians in Houston, Texas, USA).

**1000Genomes:** 
We used 1000Genomes data for the following super populations: 
AFR = ASW (African Ancestry in Southwest US), ACB (African Caribbean in Barbados), ESN (Esan in Nigeria), GWD (Gambian in Western Division, The Gambia), LWK (Luhya in Webuye, Kenya), MSL (Mende in Sierra Leone), and YRI (Yoruba in Ibadan, Nigeria).
EAS = CDX (Chinese Dai in Xishuangbanna, China), CHB (Han Chinese in Bejing, China), JPT (Japanese in Tokyo, Japan), KHV (Kinh in Ho Chi Minh City, Vietnam), and CHS (Southern Han Chinese, China).
EUR = GBR (British in England and Scotland), FIN (Finnish in Finland), IBS (Iberian populations in Spain), TSI (Toscani in Italy), and CEU (Utah residents with Northern and Western European ancestry).
SAS = BEB (Bengali in Bangladesh), GIH (Gujarati Indian in Houston,TX), ITU (Indian Telugu in the UK), PJL (Punjabi in Lahore,Pakistan), and STU (Sri Lankan Tamil in the UK).

More information can be found here: http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.populations.tsv

### Outlier Detection and Ancestry Assignment Analysis:
**Flexibility in Reference Population Selection:** A key feature of the function is its versatility in handling various reference populations. Users have the freedom to select a specific reference population against which the study samples' ancestry labels are compared. For instance, setting the parameter `outlierOf` to EUR (i.e., `Ref_EUR`) enables the identification of non-European outliers in a European context. Other reference populations like AFR (African), EAS (East Asian), and SAS (South Asian) can also be used, allowing for a wide range of comparative ancestral analyses. This flexibility ensures that `AncestryCheck` can cater to diverse research needs and objectives.

**Setting Outlier/Prediction Thresholds:** The function allows users to define specific thresholds for outlier detection/ancestry assignement, adding a layer of customization to the analysis. It employs a methodical approach by calculating PC1 and PC2 Z-scores for each subject in relation to mean and standard deviation of PC1 and PC2 for each reference population. Samples with PC1 and PC2 Z-scores within the user-defined threshold are predicted as having the same ancestry as that reference population. For example, in the case of European reference samples, the function computes the mean and standard deviaiton of the first two principal components (`mean(PC1 of Ref_EUR)`, `sd(PC1 of Ref_EUR`), `mean(PC2 of Ref_EUR)`, `sd(PC2 of Ref_EUR)`). It then calculates Z-scores for PC1 and PC2 (`Z-score PC1 of Ref_EUR <- (PC1 - mean(PC1 of Ref_EUR))/sd(PC1 of Ref_EUR)`, `Z-score PC2 of Ref_EUR <- (PC2 - mean(PC2 of Ref_EUR))/sd(PC2 of Ref_EUR)` ) for the samples in relation to the European Reference. Any study sample whose PC1 and PC2 Z-score is considered an outlier of the European population. This approach is particularly effective in pinpointing samples with ancestral backgrounds that significantly deviate from the selected reference population, providing a clear and quantifiable method to identify and analyze outliers.

### Visualization and Reporting:
**PCA Plot Visualization:** One of the most insightful aspects of `AncestryCheck` is its ability to visually present the results of the PCA. This is done through a color-coded scatter plot that displays the first two principal components. This graphical representation is instrumental in highlighting the genetic similarities and differences among the analyzed populations. By plotting individuals on this scatter plot, researchers can easily discern the underlying genetic structure and observe how study samples cluster with or diverge from various reference populations.

**Resultant Dataframe:** Following the PCA and outlier analysis, `AncestryCheck` generates a dataframe that lists the identified outlier samples. This output is particularly useful for researchers, as it provides a clear and organized view of the samples that significantly differ genetically from the chosen reference population. This information is crucial for further detailed investigations and interpretations of genetic ancestry.

**Comprehensive Reporting:** In addition to the visual and tabular outputs, the function also produces detailed reports covering various aspects of the analysis. These reports include information on allele flips and a thorough breakdown of the outlier analysis. Such comprehensive reporting enhances the overall understanding of the genetic structure and ancestry estimations, providing a multi-dimensional view of the results. This feature of `AncestryCheck` is particularly valuable for researchers who require in-depth documentation and interpretation of their genetic data. 

## `AncestryCheck` parameters in brief
The ``AncestryCheck`` function takes several arguments to customize its operations:

- `DataDir`: Directory containing input PLINK files.
- `ResultDir`: Directory for storing output files; defaults to a temporary directory.
- `finput`: Prefix of PLINK binary files for study samples.
- `reference`: Choice of reference population data, either 'HapMapIII_NCBI36' or 'ThousandGenome'.
- `filterSNP`: Flag to enable or disable SNP filtering.
- `studyLD and referLD`: Flags for applying LD-based filtering on study and reference data, respectively.
- `studyLD_window_size`, `studyLD_step_size`, `studyLD_r2_threshold`, `referLD_window_size`, `referLD_step_size`, `referLD_r2_threshold`: Parameters for LD pruning.
- `highLD_regions`: Dataframe of known high LD regions.
- `study_pop`: Dataframe containing sample IDs and ancestry labels for study samples.
- `outlier`: Flag to enable or disable outlier detection.
- `outlierOf`: Reference ancestry name for detecting outlier samples.
- `outlier_threshold`: Numeric threshold for outlier detection.

## Detailed parameter description
Before exploring the practical application of the `AncestryCheck` function, it is essential to understand the details of the above arguments and their implications:

- `DataDir`: This parameter requires a character string specifying the file path for the input PLINK binary files. It is essential for indicating where the study's genetic data is stored.

- `ResultDir` (default = tempdir()): This argument is a character string for the file path where all output files will be stored. If not specified, it defaults to R's temporary directory.

- `finput`: This parameter specifies the prefix of the input PLINK binary files for the study samples.

- `reference` (default = c("HapMapIII_NCBI36", "ThousandGenome")): A vector indicating the choice of reference population. The options include HapMap Phase 3 and 1000 Genomes Phase III.

- `filterSNP` (default = TRUE): Indicates whether to filter out SNPs. It's set to TRUE by default but can be changed to FALSE if direct joining of study and reference samples is possible.

- `studyLD` (default = TRUE): This Boolean value determines whether LD-based filtering is applied to study genotype data.

- `studyLD_window_size` (default = 50): An integer defining the window size in variant count or kilobase for LD-based filtering of the variants for the study data.

- `studyLD_step_size` (default = 5): Specifies the variant count to shift the window at the end of each step for LD filtering of the study data.

- `studyLD_r2_threshold` (default = 0.02): A numeric value between 0 and 1 setting the pairwise r2 threshold for LD-based filtering of the study data.

- `referLD` (default = FALSE): Determines whether LD-based filtering is applied to reference genotype data.

- `referLD_window_size` (default = 50), referLD_step_size (default = 5), referLD_r2_threshold (default = 0.02): These parameters define the window size, step size, and r2 threshold for LD filtering of the reference data, similar to the study data settings.

- `highLD_regions`: This parameter requires a dataframe containing known high LD regions. A dataframe from Anderson et. al.[4], 'highLD_hg19' from hg19 genome built is provided in the package.

- `study_pop`: A dataframe containing two columns for the study: sample ID (IID) and the ancestry label. An example dataframe prefied as 'example_data_study_sample_ancestry' for 'GXwasR_example' data.

- `outlier` (default = FALSE): A Boolean value specifying whether outlier detection will be performed.

- `outlierOf` (default = "EUR"): This parameter specifies the reference ancestry name for detecting outlier samples.

- `outlier_threshold` (default = 3): Specifies the numeric threshold for outlier detection. This threshold is multiplied with the Euclidean distance from the center of the PC1 and PC2 to the maximum Euclidean distance of the reference samples to determine outliers.

These parameters and their default values allow for a significant degree of customization in the `AncestryCheck` function, enabling users to tailor the analysis to their specific research needs.


## AncestryCheck Usage using HapMap 3 data as reference.

The function can be used in the following manner with our study data and reference data as HapMapIII:

```{r}
# Define parameters
library(GXwasR)
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir <- tempdir()
finput <- "GXwasR_example"
reference <- "HapMapIII_NCBI36"
data("GXwasRData")
highLD_regions <- highLD_hg19
study_pop <- example_data_study_sample_ancestry
studyLD_window_size = 50
studyLD_step_size = 5
studyLD_r2_threshold = 0.02
filterSNP = TRUE
studyLD = TRUE
referLD = TRUE
referLD_window_size = 50
referLD_step_size = 5
referLD_r2_threshold = 0.02
outlier = TRUE
outlier_threshold = 3

# Call the AncestryCheck function
source("/projects/b1137/BBose/GXwasR/R/GXwasR_main_functions.R")
source("/projects/b1137/BBose/GXwasR/R/GXwasR_helper_functions.R")
source("/projects/b1137/BBose/Panscan_codes/AncestryFunction2024_Edited._v2R.R")
ancestry_results <- AncestryCheck(
    DataDir = DataDir,
    ResultDir = ResultDir,
    finput = finput,
    reference = reference,
    filterSNP = TRUE,
    studyLD = TRUE,
    studyLD_window_size = 50,
    studyLD_step_size = 5,
    studyLD_r2_threshold = 0.02,
    referLD = FALSE,
    referLD_window_size = 50,
    referLD_step_size = 5,
    referLD_r2_threshold = 0.02,
    highLD_regions = highLD_regions,
    study_pop = study_pop,
    outlier = TRUE,
    outlierOf = "EUR",
    outlier_threshold = 3
)

Samples_with_predicted_ancestry <- ancestry_results[["Samples_with_predicted_ancestry"]]

outlier <- ancestry_results[["Outlier_samples"]]

outlier

Samples_with_predicted_ancestry[1:5,]
```

The plot shows the distribution of individual genetic samples along the first two principal components (PC1 and PC2). Each point represents a genetic sample, and they are color-coded based on reference populations (`Ref_ASIAN`, `Ref_CEU`, and `Ref_YRI`) and study populations (`Study_EUR_FIN`, `Study_EUR_GBR`, and `Study_EUR_TSI`).

The points are spread across the PCA plot, with each cluster likely representing genetic similarity within a group and differences between the groups. The reference populations, `Ref_AFR`, `Ref_EAS`, `Ref_EUR`, and `Ref_SAS` clustered distinctively, suggesting clear genetic differentiation between these populations. The study populations overlap with the `Ref_EUR` (`Euoprean`) reference population, indicating European ancestry, with each study subgroup (`FIN, GBR, TSI`) forming subclusters, suggesting finer-scale differences in ancestry within Europe. 

Since the study dataset was simulated from CEU 1000 Genome dataset, it is not suprising that we do see some outlier samples. Specifically, TSI population is not highly similar to CEU. 

## AncestryCheck Usage using 1000 genome data as reference

The function can be used in the following manner with our study data and reference data as ThousandGenome:

```{r}
# Define parameters
finput <- "GXwasR_example"
reference <- "HapMapIII_NCBI36"
library(GXwasR)
data("GXwasRData")
highLD_regions <- highLD_hg19 #"high-LD-regions-hg19-GRCh37.txt" # four columns, chr, start, end, index
study_pop <- example_data_study_sample_ancestry #PreimputeEX
studyLD_window_size = 50
studyLD_step_size = 5
studyLD_r2_threshold = 0.02
filterSNP = TRUE
studyLD = TRUE
referLD = TRUE
referLD_window_size = 50
referLD_step_size = 5
referLD_r2_threshold = 0.02
outlier = TRUE
outlier_threshold = 5
outlierOf = "EUR"

source("/projects/b1137/BBose/GXwasR/R/GXwasR_main_functions.R")
source("/projects/b1137/BBose/GXwasR/R/GXwasR_helper_functions.R")
source("/projects/b1137/BBose/Panscan_codes/AncestryFunction2024_Edited._v2R.R")
x <- AncestryCheck(DataDir = DataDir, ResultDir = ResultDir, finput = finput, reference = reference, highLD_regions = highLD_regions, study_pop = study_pop, studyLD = studyLD, referLD = referLD, outlierOf = "EUR", outlier = outlier, outlier_threshold = outlier_threshold )

Samples_with_predicted_ancestry <- ancestry_results[["Samples_with_predicted_ancestry"]]

outlier <- ancestry_results[["Outlier_samples"]]

outlier

Samples_with_predicted_ancestry[1:5,]
```

The lack of outlier samples—those that would be significantly distanced from the reference group—suggests homogeneity in the ancestry of the study population. Given that our study data was simulated based on the EUR population from the 1000 Genomes project data, the absence of outliers aligns with expectations.

The precise clustering and absence of outliers not only reaffirm the simulation parameters but also shows the reliability of the ancestry estimation methods employed. Such accuracy is crucial in genetic studies, as it reinforces confidence in the PCA's ability to discern and delineate ancestry with precision.

## Conclusion

The combination of genotype datasets from study populations with comprehensive reference datasets, followed by PCA, provides a nuanced view of ancestry. The GXwasR package simplifies this process, offering researchers and enthusiasts a robust tool for ancestry estimation. This efficient approach not only enhances our understanding of genetic diversity but also strengthens our connection to our ancestral past.

## References
(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.

(2) The International HapMap 3 Consortium. Integrating common and rare genetic variation in diverse human populations. Nature 467, 52–58 (2010). https://doi.org/10.1038/nature09298

(3) The 1000 Genomes Project Consortium. A global reference for human genetic variation. Nature 526, 68–74 (2015). https://doi.org/10.1038/nature15393

(4) Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP, Zondervan KT. Data quality control in genetic case-control association studies. Nat Protoc. 2010 Sep;5(9):1564-73. doi: 10.1038/nprot.2010.116. Epub 2010 Aug 26. PMID: 21085122; PMCID: PMC3025522.

## Citation
If you are using GXwasR package and/or following this pipeline in your study, please cite it as:
'TBD'



                                  ******************* THANK YOU! *************************

