---
title: "Genetic Correlation using GWAS Summary Statistics."
author: "Banabithi Bose"
date: "`r Sys.Date()`"
output:
  rmdformats::robobook:
  thumbnails: true
self_contained: true
gallery: true
lightbox: true
number_sections: true
toc_depth: 6
highlight: zenburn
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## GXwasR function SumstatGenCorr()

**SumstatGenCorr: Genetic Correlation Calculation from GWAS Summary Statistics**

This function calculates the genetic correlation between two summary statistics
using a specified reference LD matrix from the UK Biobank or Hapmap2 following Ning Z, Pawitan Y, Shen X (2020). High-definition likelihood inference of genetic correlations 
across human complex traits, Nature Genetics.

The function utilizes the precomputed eigenvalues and eigenvectors of LD correlation matrices for European ancestry population. 

These are the LD matrices and their eigen-decomposition from three different sets:

**1) UKB_imputed_hapmap2_SVD_eigen99_extraction:** 769,306 QCed UK Biobank imputed HapMap2 SNPs:
If one of your GWAS includes most of the HapMap 2 SNPs, but many SNPs (more than 1%) in the above HapMap 3 reference panel are absent, then this HapMap2 panel is more proper to be used for computing genetic correlation. The size is about 18 GB after unzipping

**2) UKB_imputed_SVD_eigen99_extraction:** 1,029,876 QCed UK Biobank imputed SNPs. The size is about 31 GB after unzipping. Although it takes more time, using the imputed panel provides more accurate estimates of genetic correlations. The reference panels with imputed SNPs are based on genotypes in UK Biobank, which were imputed to HRC and UK10K + 1000 Genomes.
Therefore if the GWAS includes most of the HapMap3 SNPs, then it is recommend using the imputed reference panel.

**3) UKB_array_SVD_eigen90_extraction:** 307,519 QCed UK Biobank Axiom Array SNPs. The size is about 7.5 GB after unzipping.

```{r}
library(GXwasR)
help(SumstatGenCorr)
```

## Running example with imputed datasets from Neale's lab with UKB_imputed_hapmap2_SVD_eigen99_extraction
Birth weight and type 2 diabetes based on the summary statistics with around 20,000 individuals.
```{r}
source("/projects/b1137/BBose/GXwasR/R/GXwasR_main_functions.R")
source("/projects/b1137/BBose/GXwasR/R/GXwasR_helper_functions.R")
source("/projects/b1137/BBose/GXwasR/R/utils-pipe.R")

ResultDir <- path.expand("~/b1137/BBose/MetaGWAS")
sumstat1 <- readRDS("~/b1137/BBose/MetaGWAS/gwas1.imputed.example.rds")
sumstat2 <- readRDS("~/b1137/BBose/MetaGWAS/gwas2.imputed.example.rds")
referenceLD = "UKB_imputed_hapmap2_SVD_eigen99_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=sumstat1, sumstat2=sumstat2)
```

## Running example with our D1 and DBgap's panscan I summary statistics with UKB_imputed_hapmap2_SVD_eigen99_extraction
```{r}
load("/projects/b1137/BBose/Panscan_ResultDir/ResultDir_Imputed_DG_98846_1_H55_c1_panscan_v3/FMcomb02/ResultGXwas.rda")
D1 <- ResultGXwas[ResultGXwas$TEST == "ADD",]
D1 <- D1[,c(2,4,6,7,8,12,11)]
colnames(D1) <- c("SNP","A1","N","b","se","P","Z")
D1 <- na.omit(D1)
bim1 <- read.table("/projects/b1137/BBose/Panscan_DataDir/DataDir_DG_98846_1_H55_c1_panscan_v3/PostimputeFinal.bim")[,c(2,5,6)]
colnames(bim1)<- c("SNP","A1","A2")
data1 <- merge(bim1,D1,by = c("SNP","A1"))

D2 = read.table(gzfile("~/projects/b1137/raw_data/dbGaP/GWAS/98846_1/PhenoGenotypeFiles/RootStudyConsentSet_phs000206.CGEMS_PanScan.v5.p3.c1.CADM/AnalysisFiles/New/release/panscan_analysis_CEU.txt"),sep="\t",header = T)
D2$N <- D2$Genotype.1.1.Case + D2$Genotype.1.2.Case+D2$Genotype.2.2.Case+D2$Genotype.1.1.Control+D2$Genotype.1.2.Control+D2$Genotype.2.2.Control
data2 <- D2[,c(3,4,5,14,17,18,19)] 
colnames(data2) <- c("SNP","A1","A2","b","Z","P","N")
data2$b <- log(data2$b)  # Assuming 'data$OR' holds the Odds Ratios
# Assuming your dataframe is named 'data' and has columns 'BETA' and 'TS' for Z-score
data2$se <- abs(data2$b / data2$Z)
data2 <- data2[,c("SNP", "A1", "A2", "N", "b", "se", "P","Z")]
data2 <- na.omit(data2)

sumstat1 <- data1
sumstat2 <- data2
referenceLD = "UKB_imputed_hapmap2_SVD_eigen99_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=sumstat1, sumstat2=sumstat2)
```

## Our D1 and Neale's lab's weight data with UKB_imputed_hapmap2_SVD_eigen99_extraction
```{r}
sumstat1 <- data1
sumstat2 <- readRDS("~/b1137/BBose/MetaGWAS/gwas1.imputed.example.rds")
referenceLD = "UKB_imputed_hapmap2_SVD_eigen99_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=sumstat2)
```

## Try our D1 and D2 with referenceLD UKB_imputed_SVD_eigen99_extraction
```{r}
sumstat1 <- data1
sumstat2 <- data2
referenceLD = "UKB_imputed_SVD_eigen99_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=data2)
```
**More SNPs seems better.**

## Our D1 and Neale's lab's birth weight data with UKB_imputed__SVD_eigen99_extraction
```{r}
sumstat1 <- data1
sumstat2 <- readRDS("~/b1137/BBose/MetaGWAS/gwas1.imputed.example.rds")
referenceLD = "UKB_imputed_SVD_eigen99_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=sumstat2)
```

## Try our D1 and D2 with referenceLD UKB_array_SVD_eigen90_extraction
```{r}
referenceLD = "UKB_array_SVD_eigen90_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=data2)
```
**Interestingly, heritability of trait 1 seems closer to true heritability. Is it because Axiom array is high quality and only genotyped on EUR unlike Hapmap and 1000Genome?**


#Should genetic correlation between same two summary statistics is ~ 1?

## What happened when I tried our data1 with data1 with referenceLD UKB_imputed_hapmap2_SVD_eigen99_extraction

```{r}
referenceLD = "UKB_imputed_hapmap2_SVD_eigen99_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=data1)
```

## What happened when I tried our data1 with data1 with referenceLD UKB_imputed_SVD_eigen99_extraction
```{r}
referenceLD = "UKB_imputed_SVD_eigen99_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=data1)
```
**Here, we have a high number of SNPs but the algorithm did not converge.**

## What happened when I tried our data1 with data1 with referenceLD UKB_array_SVD_eigen90_extraction
With Axiom array the computation of heritability is near to true heritability for our D1.

```{r}
referenceLD <- "UKB_array_SVD_eigen90_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=data1)
```
**Great! We have a high correlation but not "1". Why high: Maybe because of higher quality SNPs in Axiom arrar.
Why not 1: Maybe because of small sample size or underpowered GWAS.**

## Will the estimation improve if we use Axiom array for data1 and Neale's lab data
```{r}
sumstat2 <- readRDS("~/b1137/BBose/MetaGWAS/gwas1.imputed.example.rds")
referenceLD <- "UKB_array_SVD_eigen90_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=sumstat2)
```
**Here for Neale's lab data the overlap is only 40%. So the result is not surprising.**


## Try HDL provided small example data (20,000 SNPs) with referenceLD UKB_array_SVD_eigen90_extraction
```{r}
load("~/gwas1.example.rda")
load("~/gwas2.example.rda")
referenceLD = "UKB_array_SVD_eigen90_extraction"
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=gwas1.example, sumstat2=gwas2.example)
```
**Small number of SNPs anyways not good.**

**Note:** Singularities or ill-conditioned matrices can occur due to **collinear SNPs**, insufficient variation, or extremely small values, causing numerical instability and non-convergence.

# HDL decoded
Checking the provided likelihood functions (**llfun and llfun.gcov.part.2**), it's possible that very similar summary statistics (e.g., identical Z-scores) for two traits can lead to convergence issues in this algorithm due to several reasons:

**1. Collinearity and Identical Data:**

If the summary statistics are identical between two traits:

**Collinearity:** The covariance matrix may become singular or nearly singular, which can make the optimization unstable.

Example snippet where identical Z-scores may cause collinearity issues

llfun.gcov.part.2 <- function(param, h11, h22, rho12, M, N1, N2, N0, Nref,
                              lam0, lam1, lam2, bstar1, bstar2, lim = exp(-10)){
   # Estimating covariance matrices
   lam11 = h11[1] / M * lam1^2 - h11[1] * lam1 / Nref + h11[2] * lam1 / N1
   lam22 = h22[1] / M * lam2^2 - h22[1] * lam2 / Nref + h22[2] * lam2 / N2
   lam12 = if (N0 > 0) h12 / M * lam1 * lam2 + N0 / (N1 * N2) * int * lam1 else h12 / M * lam1 * lam2

   # High collinearity leads to potential singularity
   if (is.singular(matrix(c(lam11, lam12, lam12, lam22), nrow = 2))) {
       stop("Singular matrix due to collinearity")
   }
}


**Identical Statistics:** The functions rely on variability between the traits, and identical statistics can lead to a flat or undefined likelihood surface, causing numerical instability.

Example where identical summary stats lead to flat likelihood surface

llfun <- function(param, N, M, Nref = 1000, lam, bstar, lim = exp(-10)){
   h2 = param[1]
   int = param[2]
   lamh2 = h2 / M * lam^2 - h2 * lam / Nref + int * lam / N

   if (all(bstar == bstar[1])) {
       stop("Identical statistics detected, leading to undefined likelihood surface")
   }

   lamh2 = ifelse(lamh2 < lim, lim, lamh2)
   ll = sum(log(lamh2)) + sum(bstar^2 / (lamh2))
   return(ll)
}


**2. Flat Likelihood Surface:**

If the data is identical, the likelihood surface might be flat, meaning that the algorithm has difficulty finding a unique solution due to lack of gradients.
In llfun.gcov.part.2, the covariance (h12) may end up zero or undefined if there's no variance in the data.

**3. Optimization Method Sensitivity:**

The numerical optimization ('optim') used for finding the best fit is sensitive to data quality.
Identical data can result in values that don't fit the assumptions or cause overflows in likelihood calculations. This may results in **Algorithm failed to converge after trying different initial values.** 

**4. Variance Estimates:**

The variance estimates (lamh2, lam22.1) are sensitive to the differences in summary statistics.
Identical data may result in variance estimates being too small or large, leading to numerical instability in the log-likelihood calculations.

## Why panel UKB_imputed_SVD_eigen99_extraction does not converge but UKB_array_SVD_eigen90_extraction converge for the same data1?

**Expected Genetic Correlation:**
When using identical summary statistics, the genetic correlation is theoretically expected to be very close to 1, as the traits are the same.

However, this can vary in practice due to the quality of the data and the specific nuances of the LD reference panel.

**Sensitivity to LD Reference Panels:**
Different LD panels can have distinct effects on the computation, especially in cases of identical summary statistics:

**LD Structure:** Panels with better-defined LD structures (like the Axiom array) may produce more stable covariance matrices, leading to better convergence.

**Imputation Accuracy:** LD panels with more imputed SNPs (like the HRC, 1000 Genomes, HapMap3 panel) may introduce noise, making convergence more challenging.

**Effect of Imputation and Rare Variants:**
Imputed panels often have a larger number of SNPs, including rare variants, which can introduce noise and increase the complexity of the covariance structure.

The additional noise and potential artifacts from imputation could make it difficult for the algorithm to converge, especially when working with identical summary statistics.

**Impact of Multicollinearity:**
Identical summary statistics can lead to high multicollinearity, which may be mitigated differently depending on the LD reference panel.

High multicollinearity can destabilize the covariance matrix calculations, resulting in non-convergence for one panel but not the other.

In summary, the differences in convergence for the two LD panels when computing genetic correlation with identical summary statistics highlight the importance of:

LD structure quality,
Imputation noise,
The inherent properties of the SNPs in the panels.
The Axiom array may offer a cleaner, more stable set of SNPs with well-defined LD, which is critical for achieving convergence in the genetic correlation computation.

## I tried GenomicSem package
763976 out of 769306 (99.31%) SNPs in reference panel are available in the GWAS of Data1.txt

Estimation for cell: 1 out of: 3 cells is ongoing ... 100%

Estimation for cell: 2 out of: 3 cells is ongoing ... 100%

301051 out of 769306 (39.13%) SNPs in reference panel are available in the GWAS of Data2.txt

Warning: More than 1% SNPs in reference panel are missed in the GWAS. This may generate bias in estimation.

Please make sure that you are using correct reference panel.

Estimation for cell: 3 out of: 3 cells is ongoing ... 100%Warning message:

In cov2cor(V) : diag(.) had 0 or NA entries; non-finite result is doubtful

**I thought this is happening because DBgap summary statistics is in hg18, whereas ours is in hg19, also hm3 is in hg19.**

In cov2cor(V) : diag(.) had 0 or NA entries; non-finite result is doubtful

Hence, I updated the SNPs using proper bim file but with no use.

**I realized this is happening because DBgap's summary statistics is not imputed.**

**I used .Bed file made from summary statistics** and processed via Crossmap to update the coordinates but got lower overlap between DBgap and reference panel Hapmap3 provided by the GenomicSem.

## Can we use LD pruned SNPs in this computation
**The answer is NO.**

if we use LD pruned SNPs:

32641 out of 769306 (4.24%) SNPs in reference panel are available in the GWAS of Data1.txt 

Warning: More than 1% SNPs in reference panel are missed in the GWAS. This may generate bias in estimation. Please make sure that you are using correct reference panel. 

## What happened when I tried to correlate the same dataset in GenomicSem

**It did not converge.**

If two summary statistics are exactly the same, the covariance between them will be equal to the variance of either one of the statistics. 

The presence of the same value for variance in both 𝐷1 and 𝐷2 could suggest similar traits or measurements under the same conditions but with different outcomes based on the conditions or methods of measurement leading to a negative covariance.


## Conclusion statement
**The genetic correlation analysis is sensitive to the number of SNPs in the summary statistics and the quality of the LD reference panel.**

In Panscan datasets, the result is:

Heritability of phenotype 1:  0.00e+00 (0.00e+00) 

Heritability of phenotype 2:  0.00e+00 (0.00e+00) 

Genetic Covariance:  -0.1536 (0.1436) 

Genetic Correlation:  -Inf (NA) 

**About Warning: Heritability of one trait was estimated to be 0, which may due to:**

          1) The true heritability is very small;
          
          2) The sample size is too small;
          
          3) Many SNPs in the chosen reference panel misses in the GWAS.
          
          4) There is a severe mismatch between the GWAS population and the population for computing reference panel

**Note:** True heritability in pancreatic cancer is 21%. Summary statistics may not give this estimate.

**Potential Calculation Issues**

  **HDL Package:** 
   It uses likelihood-based methods which can also run into numerical difficulties with very low or zero heritability data, possibly leading to non-finite results.

**Relevant Sections in the HDL Algorithm**:

 **Heritability Calculation:**

The heritability of each trait is calculated using linear models (lm()) on squared z-scores against LD scores.
Later, likelihood-based methods (optim()) refine these estimates. **Here, having small number of SNPs with a few significant hits may lead to low to no variation that leads towards zero heritability estimate.**

  **Genetic Covariance:**

Genetic covariance between traits is calculated using a linear model (lm()) on the product of z-scores.

  **Genetic Correlation:**

The genetic correlation (rg) is derived by dividing the genetic covariance by the square root of the product of heritability values: The genetic correlation (\(r_g\)) can be calculated as shown below:

$$
r_g = \frac{h_{12}}{\sqrt{h_{11} \cdot h_{22}}}
$$

Where:
- \(h_{11}\) is the heritability of Trait 1
- \(h_{22}\) is the heritability of Trait 2
- \(h_{12}\) is the genetic covariance between the two traits

**Impact of Zero Heritability**

  **Zero Heritability:**
   
   If the heritability for one of the traits is zero, it implies that genetic factors don't explain any variance in that trait.

  **Mathematical Impact:**
   
   In the calculation for h11 or h22 is zero, the denominator becomes zero.
This results in division by zero, which in mathematical operations typically yields NaN, Inf, or -Inf.

Variance-Covariance Matrix from summary statistics D1 and D2:
```{r}
knitr::include_graphics("Picture1.png")
```


**Analysis and Impact**

The number of SNPs and their overlap with the reference LD panel and the **quality** of the reference significantly impact genetic correlation computation.

**What is the best LD panel:** Best LD reference panel for Panscan seems to be UKB_array_SVD_eigen90_extraction due to accurate heritability estimate. This UK Biobank Axiom Array has extensive genomic coverage, including common and rare variants than hapmap or 1000 genome. The quality of the SNPs is higher in Axiom array. Also it is meant for only EUR in UKBB.

**Number of SNPs:**

More SNPs in the summary statistics generate more variation, leading to more accurate results.
LD Reference Panel: The choice of the LD reference panel is crucial for accurate computation, as the algorithm is sensitive to **LD information quality** and **overlap** with the GWAS data.

**Other factors:**

**1. Difficulty in Detecting Genetic Associations**

The combination of a **small sample size** and low heritability makes it challenging to detect genetic associations accurately. Genetic factors may only account for a small portion of the trait's variability, which is further overshadowed by environmental factors or measurement errors. This challenge becomes more pronounced with the reliance on summary statistics.

**2. Larger Sample Sizes and Imputed Summary Statistics are Required**

Increasing the sample size is necessary to enhance statistical power, making it easier to detect even minor genetic influences on pancreatic cancer. However, the limited sample size in the current dataset hinders this approach, making it challenging to achieve sufficient power for reliable results.

**3. Statistical Significance and Noise**

With a limited sample size, the signal-to-noise ratio in the data is diminished. This makes it difficult to distinguish true genetic effects from random fluctuations, leading to a higher likelihood of false positives (type I errors) or missing true effects (type II errors). This instability is further exacerbated by using summary statistics.

**4. Polygenic Traits**

Pancreatic cancer's genetic basis likely involves many genes with small effects, underscoring its polygenic nature. This dilution of effect sizes complicates the identification of significant genetic variants in underpowered datasets.

**5. Impact on Heritability Estimates**

The small sample size and low heritability bias the estimates, and the genetic architecture of pancreatic cancer includes non-additive factors like gene-gene and gene-environment interactions.

**6. Meta-Analyses and Collaborative Efforts**

Given the difficulties with small sample size and low heritability, GWAS on pancreatic cancer often require meta-analyses and collaborations to pool data and resources, boosting statistical power and reliability.

## Summary

In summary, the genetic correlation analysis is highly sensitive to the number of SNPs and the quality of the LD reference panel. More SNPs in the summary statistics provide more variation and yield better results, while good overlap with the reference LD panel enhances the quality of genetic correlation computations. Also, good sample size is required to have power in this estimation.

## Try our D1D2D4D7malefixed and D1D2D4D7femalefixed with referenceLD UKB_imputed_SVD_eigen99_extraction
```{r}
load("~/b1137/BBose/MetaGWAS/MResultDir/D1D2D4D7malefixed.Rda")
load("~/b1137/BBose/MetaGWAS/MResultDir/D1D2D4D7femalefixed.Rda")
D1D2D4D7malefixed$N <- 3688+3504  # All samples from meta-analysis
sumstat1 <- D1D2D4D7malefixed[,c(3,4,13,9,10,8)]
colnames(sumstat1) <- c("SNP","A1","N","b","se","P")
sumstat2 <- data2
referenceLD = "UKB_imputed_SVD_eigen99_extraction"
#c("SNP","A1","N","b","se","P","Z")
res <- SumstatGenCorr(ResultDir=ResultDir, referenceLD =referenceLD, sumstat1=data1, sumstat2=data2)
```


