<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Banabithi Bose, Northwestern University" />


<title>GXwasR Functions</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">GXwasR Functions</h1>
<h4 class="author">Banabithi Bose, Northwestern University</h4>



<div id="gxwasr-description" class="section level1">
<h1><a id="GXwasR Description"></a> GXwasR Description:</h1>
<p>This package implements various statistical genetics models for
Genome-Wide Association (GWA) and X-Chromosome Wide Association (XWA)
analyses in a sex-combined or sex-stratified way considering
X-Chromosome Inactivation (XCI) pattern. In addition to association
analysis, the package also enables testing for sex differences in
genetic effects, including the implementation of specific models and
applying best practices for additional quality control (QC) of genetic
data required for these tests. The package includes thirty different
functions in six different categories (A-F) which enable a comprehensive
pipeline for sex-aware genetic association analysis of common variants
with unrelated individuals.</p>
</div>
<div id="functions-overview" class="section level1">
<h1>Functions Overview</h1>
<p>This document lists all the functions in GXwasR package.It offers
thirty-three distinct functions, which are organized into six main
categories: A) Pre-imputation QC, B) Post-imputation QC, C) Sex-combined
and sex-stratified GWAS with specialized analysis for XWAS, D)
Sex-differential test, E) High level analysis and a range of F) Utility
Functions. These categories and their respective functions are detailed
in the below Table.</p>
<div id="table-of-functions" class="section level2">
<h2>Table of Functions</h2>
<table>
<colgroup>
<col width="10%" />
<col width="67%" />
<col width="22%" />
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
<th>Category</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>QCsnp()</td>
<td>Performs quality control for SNPs in PLINK binary files.</td>
<td>Pre-imputation QC, Post-imputation QC</td>
</tr>
<tr class="even">
<td>QCsample()</td>
<td>Identifies outlier individuals based on heterozygosity and missing
genotype rates.</td>
<td>Pre-imputation QC, Post-imputation QC</td>
</tr>
<tr class="odd">
<td>AncestryCheck()</td>
<td>Evaluates samples’ ancestry and flags outliers using PCA.</td>
<td>Pre-imputation QC</td>
</tr>
<tr class="even">
<td>SexCheck()</td>
<td>Compares sex assignments with predictions from X chromosome
inbreeding coefficients.</td>
<td>Pre-imputation QC</td>
</tr>
<tr class="odd">
<td>Xhwe()</td>
<td>Filters X-chromosome variants violating Hardy-Weinberg Equilibrium
in females.</td>
<td>Post-imputation QC</td>
</tr>
<tr class="even">
<td>MAFdiffSexControl()</td>
<td>Tests for significant MAF differences between sexes in control
samples.</td>
<td>Post-imputation QC</td>
</tr>
<tr class="odd">
<td>FilterRegion()</td>
<td>Filters out specific chromosomal regions from input PLINK
files.</td>
<td>Post-imputation QC, Utility Functions</td>
</tr>
<tr class="even">
<td>GXwas()</td>
<td>Runs GWAS models in autosomes with XWAS models like “FMcomb01”,
“FMcomb02”, “FMstratified” for binary and quantitative traits, and
“GWAScxci” for binary traits. Focuses on additive SNP effects,
multi-collinearity issues, and includes multiple covariates and their
interactions.</td>
<td>Sex-combined and sex-stratified GWAS with XWAS</td>
</tr>
<tr class="odd">
<td>PvalComb()</td>
<td>Combines p-values from separate GWAS using various statistical
methods and corrects summary p-values.</td>
<td>Sex-combined and sex-stratified GWAS with XWAS</td>
</tr>
<tr class="even">
<td>SexDiff()</td>
<td>Evaluates sex differences in genetic effect size for each SNP.</td>
<td>Sex-differential test</td>
</tr>
<tr class="odd">
<td>SexDiffZscore()</td>
<td>Presumably analyzes sex differences using Z-score methodology,
comparing genetic effect sizes between males and females.</td>
<td>Sex-differential test</td>
</tr>
<tr class="even">
<td>DiffZeroOne()</td>
<td>Assesses Z-scores for deviation from one and zero for statistics
like genetic correlation.</td>
<td>Sex-differential test</td>
</tr>
<tr class="odd">
<td>TestXGene()</td>
<td>Performs gene-based association tests using GWAS/XWAS summary
statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>MetaGXwas()</td>
<td>Combines summary-level GWAS results using fixed-effect and
random-effect models.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>ComputePRS()</td>
<td>Calculates polygenic risk scores from GWAS summary statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>GeneticCorrBT()</td>
<td>Computes genetic correlation between to traits.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>EstimateHerit()</td>
<td>Computes SNP heritability using GREML or LDSC models.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>SexRegress()</td>
<td>Not previously described. Presumably involves regression analyses
specific to sex-stratified data.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>FilterPlinkSample()</td>
<td>Prepares PLINK binary files with desired samples based on specified
criteria.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>ComputeGeneticPC()</td>
<td>Computes principal components from a genetic relationship matrix for
population stratification correction.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>ClumpLD()</td>
<td>Performs linkage disequilibrium clumping of SNPs.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>GetMFPlink()</td>
<td>Prepares separate male and female PLINK binary files from combined
files.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>plinkVCF()</td>
<td>Converts VCF files to plink binary formats and vice versa, including
creation of dummy FAM files.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>MergeRegion()</td>
<td>Combines two genotype datasets based on common or all SNPs.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>FilterAllele()</td>
<td>Filters out multi-allelic variants from the genetic dataset,
essential for maintaining dataset integrity and simplifying genetic
analyses.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>PlinkSummary()</td>
<td>Provides a summary of genotype datasets in plink format.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>FilterSNP()</td>
<td>Filters out specific SNPs from the dataset based on user-defined
criteria.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>DummyCovar()</td>
<td>Recodes a categorical covariate into binary dummy variables for
statistical analysis in GXwasR.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>GXWASmiami()</td>
<td>Generates Miami plots for GWAS and XWAS.</td>
<td></td>
</tr>
<tr class="even">
<td>Utility Functions</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Download_reference()</td>
<td>Download Hapmap phase 3 and 1000 Genome phase 3 Data.</td>
<td></td>
</tr>
<tr class="even">
<td>Utility Functions</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>SumstatGenCorr()</td>
<td>Genetic Correlation Calculation from GWAS Summary Statistics.</td>
<td></td>
</tr>
<tr class="even">
<td>High level analysis</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>LDPrune()</td>
<td>Performs LD pruning on SNP data.</td>
<td></td>
</tr>
<tr class="even">
<td>Utility Functions</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>executePlinkMAF()</td>
<td>calculate minor allele frequencies.</td>
<td></td>
</tr>
<tr class="even">
<td>Utility Functions</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(GXwasR)</span></code></pre></div>
<pre><code>## 
## GXwasR: Genome-wide and x-chromosome wide association analyses applying
## best practices of quality control over genetic data</code></pre>
<pre><code>## Version 0.7.0 () installed</code></pre>
<pre><code>## Author: Banabithi Bose [aut, cre] (&lt;https://orcid.org/0000-0003-0842-8768&gt;)</code></pre>
<pre><code>## Maintainer: Banabithi Bose &lt;banabithi.bose@gmail.com&gt;</code></pre>
<pre><code>## Tutorial: https://github.com</code></pre>
<pre><code>## Use citation(&quot;GXwasR&quot;) to know how to cite this work.</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">library</span>(printr)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by &#39;printr&#39;:
##   method                from     
##   knit_print.data.frame rmarkdown</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">library</span>(rmarkdown)</span></code></pre></div>
</div>
</div>
<div id="function-1" class="section level1">
<h1>Function 1</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">help</span>(QCsnp, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>QCsnp</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="QCsnp">QCsnp: Quality control (QC) for SNPs.</h2>

<h3>Description</h3>

<p>This function performs quality control on genetic variant data from PLINK binary files. It filters variants based on parameters like minor allele frequency, Hardy-Weinberg equilibrium, call rate, and differential missingness between cases and controls. It also handles monomorphic SNPs and applies linkage disequilibrium-based filtering if specified.
</p>


<h3>Usage</h3>

<pre><code class="language-R">QCsnp(
  DataDir,
  ResultDir = tempdir(),
  finput,
  foutput,
  casecontrol = TRUE,
  hweCase = NULL,
  hweControl = NULL,
  hwe = NULL,
  maf = 0.05,
  geno = 0.1,
  monomorphicSNPs = FALSE,
  caldiffmiss = FALSE,
  diffmissFilter = FALSE,
  dmissX = FALSE,
  dmissAutoY = FALSE,
  highLD_regions = NULL,
  ld_prunning = FALSE,
  window_size = 50,
  step_size = 5,
  r2_threshold = 0.02
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QCsnp_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="QCsnp_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="QCsnp_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files with both male and female samples. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="QCsnp_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files if the filtering option for the SNPs is chosen. The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="QCsnp_:_casecontrol">casecontrol</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; indicating if the input plink files has cases-control status or not. The default is FALSE.</p>
</td></tr>
<tr><td><code id="QCsnp_:_hweCase">hweCase</code></td>
<td>
<p>Numeric value between 0 to 1 or NULL for removing SNPs which fail Hardy-Weinberg equilibrium for cases. The default is NULL.</p>
</td></tr>
<tr><td><code id="QCsnp_:_hweControl">hweControl</code></td>
<td>
<p>Numeric value between 0 to 1 or NULL for removing SNPs which fail Hardy-Weinberg equilibrium for controls. The default is NULL.</p>
</td></tr>
<tr><td><code id="QCsnp_:_hwe">hwe</code></td>
<td>
<p>Numeric value between 0 to 1 or NULL for removing SNPs which fail Hardy-Weinberg equilibrium for entire dataset. The default is NULL.</p>
</td></tr>
<tr><td><code id="QCsnp_:_maf">maf</code></td>
<td>
<p>Numeric value between 0 to 1 for removing SNPs with minor allele frequency less than the specified threshold. The default is 0.05.</p>
</td></tr>
<tr><td><code id="QCsnp_:_geno">geno</code></td>
<td>
<p>Numeric value between 0 to 1 for removing SNPs that have less than the specified call rate. The default is 0.05. Users can set this as NULL for not applying this filter.</p>
</td></tr>
<tr><td><code id="QCsnp_:_monomorphicSNPs">monomorphicSNPs</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for filtering out monomorphic SNP. The default is &quot;TRUE&quot;.</p>
</td></tr>
<tr><td><code id="QCsnp_:_caldiffmiss">caldiffmiss</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to compute differential missingness between cases and controls for each SNP (threshold is 0.05/length(unique(No. of. SNPs in the test))). The default is TRUE.</p>
</td></tr>
<tr><td><code id="QCsnp_:_diffmissFilter">diffmissFilter</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to filter out the SNPs or only flagged them for differential missingness in cases vs contols. The deafailt is &quot;TRUE&quot;.</p>
</td></tr>
<tr><td><code id="QCsnp_:_dmissX">dmissX</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for computing differential missingness between cases and controls for X chromosome SNPs only. The default is &quot;FALSE&quot;. The diffmissFilter will work for all these SNPs.</p>
</td></tr>
<tr><td><code id="QCsnp_:_dmissAutoY">dmissAutoY</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for computing differential missingness between cases and controls for SNPs on autosomes and Y chromosome only. The default is &quot;FALSE&quot;. If dmissX and dmissAutoY are both FALSE, then this will be computed genome-wide. The diffmissFilter will work for all these SNPs.</p>
</td></tr>
<tr><td><code id="QCsnp_:_highLD_regions">highLD_regions</code></td>
<td>
<p>A dataframe with known high LD regions. A dataframe from Anderson et. al, 2010 (3) is provided with the package.</p>
</td></tr>
<tr><td><code id="QCsnp_:_ld_prunning">ld_prunning</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for applying linkage disequilibrium (LD)-based filtering.</p>
</td></tr>
<tr><td><code id="QCsnp_:_window_size">window_size</code></td>
<td>
<p>Integer value, specifying a window size in the variant counts for LD-based filtering. The default is 50.</p>
</td></tr>
<tr><td><code id="QCsnp_:_step_size">step_size</code></td>
<td>
<p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering. The default is 5.</p>
</td></tr>
<tr><td><code id="QCsnp_:_r2_threshold">r2_threshold</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering. The default is 0.02.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two objects, namely, MonomorSNPs and DiffMissSNPs containing monomorphic SNPs and SNPs with differential missingness in cases vs controls, respectively. Output plink binary files in the working directory.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R">DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
finput &lt;- &quot;GXwasR_example&quot;
foutput &lt;- &quot;Test_output&quot;
geno &lt;- NULL
maf &lt;- 0.05
casecontrol &lt;- FALSE
hweCase &lt;- NULL
hweControl &lt;- NULL
hweCase &lt;- NULL
monomorphicSNPs &lt;- FALSE
caldiffmiss &lt;- FALSE
ld_prunning &lt;- FALSE
x &lt;- QCsnp(DataDir = DataDir, ResultDir = ResultDir, finput = finput,foutput = foutput, geno = geno, maf = maf,hweCase = hweCase, hweControl = hweControl, ld_prunning = ld_prunning, casecontrol = casecontrol, monomorphicSNPs = monomorphicSNPs, caldiffmiss = caldiffmiss)
</code></pre>


</div>
</div>
</div>
<div id="function-2" class="section level1">
<h1>Function 2</h1>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">help</span>(QCsample, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>QCsample</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="QCsample">QCsample: Quality control for samples in the plink binary files.</h2>

<h3>Description</h3>

<p>This function identifies outlier individuals for heterozygosity and/or missing genotype rates, which aids in the detection of samples with subpar DNA quality and/or concentration that should be removed from the study. Individuals missing more than 3-7% of their genotype calls are often excluded (1)from the analysis.
Having the correct designation of sex is important to obtain accurate genotype rate estimates, or avoid incorrectly removing samples, etc. Details can be accessed from the paper.
</p>


<h3>Usage</h3>

<pre><code class="language-R">QCsample(
  DataDir,
  ResultDir,
  finput,
  foutput = NULL,
  imiss,
  het,
  small_sample_mod = FALSE,
  IBD,
  IBDmatrix = FALSE,
  ambi_out = TRUE,
  legend_text_size = 8,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9,
  filterSample = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QCsample_:_DataDir">DataDir</code></td>
<td>
<p>Character string, specifying the file path of the input PLINK binary files. The default is NULL.</p>
</td></tr>
<tr><td><code id="QCsample_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="QCsample_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files with both male and female samples. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="QCsample_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files if filtering option for the samples is chosen.</p>
</td></tr>
<tr><td><code id="QCsample_:_imiss">imiss</code></td>
<td>
<p>Numeric value between 0 to 1 for removing samples that have more than the specified missingness. The default is 0.03.</p>
</td></tr>
<tr><td><code id="QCsample_:_het">het</code></td>
<td>
<p>Positive numeric value, specifying the standard deviation from the mean heterozygosity rate. The samples whose rates are more than the specified sd from the mean heterozygosity rate are removed. The default is 3. With this default value, outlying heterozygosity rates would remove individuals who are three sd away from the mean rate (1).</p>
</td></tr>
<tr><td><code id="QCsample_:_small_sample_mod">small_sample_mod</code></td>
<td>
<p>Boolean value indicating whether to apply modifications for small sample sizes. Default is FALSE.</p>
</td></tr>
<tr><td><code id="QCsample_:_IBD">IBD</code></td>
<td>
<p>Numeric value for setting the threshold for Identity by Descent (IBD) analysis. Default is NULL.</p>
</td></tr>
<tr><td><code id="QCsample_:_IBDmatrix">IBDmatrix</code></td>
<td>
<p>Boolean value indicating whether to generate an entire IBD matrix. Default is FALSE. In this case filtered IBD matrix will be stored.</p>
</td></tr>
<tr><td><code id="QCsample_:_ambi_out">ambi_out</code></td>
<td>
<p>Boolean value indicating whether to process ambiguous samples. Default</p>
</td></tr>
<tr><td><code id="QCsample_:_legend_text_size">legend_text_size</code></td>
<td>
<p>Integer, specifying the size for legend text in the plot.</p>
</td></tr>
<tr><td><code id="QCsample_:_legend_title_size">legend_title_size</code></td>
<td>
<p>Integer, specifying the size for the legend title in the plot.</p>
</td></tr>
<tr><td><code id="QCsample_:_axis_text_size">axis_text_size</code></td>
<td>
<p>Integer, specifying the size for axis text in the plot.</p>
</td></tr>
<tr><td><code id="QCsample_:_axis_title_size">axis_title_size</code></td>
<td>
<p>Integer, specifying the size for the axis title in the plot.</p>
</td></tr>
<tr><td><code id="QCsample_:_title_size">title_size</code></td>
<td>
<p>Integer, specifying the size of the title of the plot heterozygosity estimate vs missingness across samples.</p>
</td></tr>
<tr><td><code id="QCsample_:_filterSample">filterSample</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for filtering out the samples or not (i.e., only flagged). The default is &quot;TRUE&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of heterogysity estimate vs missingness accross sample and a list containing five R dataframe objects, namely, HM (samples with outlying heterozygosity and/or missing genotype rates), Failed_Missingness (samples with missing genotype rates), Failed_heterozygosity (samples with outlying heterozygosity), Missingness_results (missingness results) and Heterozygosity_results (heterozygosity results) with output plink files in ResultDir if filtering out the samples option is chosen.
Missingness_results contains missingness results for each individual, with six columns as FID, IID, MISS_PHENO, N_MISS, N_GENO and F_MISS for Family ID, Within-family ID, Phenotype missing? (Y/N), Number of missing genotype call(s), not including obligatory missings or heterozygous haploids, number of potentially valid call(s), and missing call rate, respectively.
Heterozygosity_results contains heterozygosity results for each individual, with six columns as FID, IID, O(HOM), E(HOM), N(NM), and F for Family ID, Within-family ID, Observed number of homozygotes, Expected number of homozygotes, Number of (non-missing, non-monomorphic) autosomal genotype observations and, Method-of-moments F coefficient estimate, respectively.
</p>


<h3>Examples</h3>

<pre><code class="language-R">DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
finput &lt;- &quot;GXwasR_example&quot;
foutput &lt;- &quot;Test_output&quot;
imiss = 0.01
het = 2
small_sample_mod = FALSE
IBD = 0.2
IBDmatrix = FALSE
ambi_out = TRUE

x = QCsample(DataDir = DataDir,ResultDir = ResultDir, finput = finput,foutput = foutput, imiss = imiss,het = het, IBD = IBD,ambi_out = ambi_out)
</code></pre>


</div>
</div>
</div>
<div id="function-3" class="section level1">
<h1>Function 3</h1>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">help</span>(AncestryCheck, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>AncestryCheck</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="AncestryCheck">AncestryCheck: Evaluation of the samples&#39; ancestry label.</h2>

<h3>Description</h3>

<p>This function displays the result of the ancestry analysis in a color-coded scatter plot of the first two principal components for samples of the reference populations and the study population. Specifically, it compares the study samples&#39; ancestry labels to a panel representing a reference population, and it also flags the outlier samples with respect to a chosen reference population. The function first filters the reference and study data for non-A-T or G-C SNPs. It next conducts LD pruning, fixes the chromosome mismatch between the reference and study datasets, checks for allele flips, updates the locations, and flips the alleles. The two datasets are then joined, and the resulting genotype dataset is subjected to Principal Component Analysis (PCA). The detection of population structure down to the level of the reference dataset can then be accomplished using PCA on this combined genotyping panel.  For instance, the center of the European reference samples is determined using the data from principal components 1 and 2 (median(PC1 europeanRef), median(PC2 europeanRef)). It determines the European reference samples&#39; maximum Euclidean distance (maxDist) from this center. All study samples that are non-European, or outliers, are those whose Euclidean distances from the center are more than or equal to the radius r= outlier threshold* maxDist. This function utilizes the HapMap phase 3 data in NCBI 36 and 1000GenomeIII in CGRCh37. Both study and reference datasets should be of the same genome build. If not, users need to lift over one of the datasets to the same build.
</p>


<h3>Usage</h3>

<pre><code class="language-R">AncestryCheck(
  DataDir,
  ResultDir = tempdir(),
  finput,
  reference = c(&quot;HapMapIII_NCBI36&quot;, &quot;ThousandGenome&quot;),
  filterSNP = TRUE,
  studyLD = TRUE,
  studyLD_window_size = 50,
  studyLD_step_size = 5,
  studyLD_r2_threshold = 0.02,
  referLD = FALSE,
  referLD_window_size = 50,
  referLD_step_size = 5,
  referLD_r2_threshold = 0.02,
  highLD_regions,
  study_pop,
  outlier = FALSE,
  outlierOf = &quot;EUR&quot;,
  outlier_threshold = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AncestryCheck_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files for the study samples.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_reference">reference</code></td>
<td>
<p>Boolean value,&#39;HapMapIII_NCBI36&#39; and &#39;ThousandGenome&#39;, specifying Hapmap Phase3 (1) and 1000 Genomes phase III (2) reference population, respectively. The default is &#39;HapMapIII_NCBI36&#39;.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_filterSNP">filterSNP</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for filtering out the SNPs. The default is TRUE. We recommend setting it FALSE only when the users are sure that they could join the study and reference samples directly.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_studyLD">studyLD</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for applying linkage disequilibrium (LD)-based filtering on study genotype data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_studyLD_window_size">studyLD_window_size</code></td>
<td>
<p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering of the variants for the study data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_studyLD_step_size">studyLD_step_size</code></td>
<td>
<p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering for the study data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_studyLD_r2_threshold">studyLD_r2_threshold</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering for the study data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_referLD">referLD</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for applying linkage disequilibrium (LD)-based filtering on reference genotype data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_referLD_window_size">referLD_window_size</code></td>
<td>
<p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering of the variants for the reference data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_referLD_step_size">referLD_step_size</code></td>
<td>
<p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering for the reference data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_referLD_r2_threshold">referLD_r2_threshold</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering for the reference data.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_highLD_regions">highLD_regions</code></td>
<td>
<p>A dataframe with known high LD regions. A dataframe from Anderson et. al, 2010 (3) is provided with the package.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_study_pop">study_pop</code></td>
<td>
<p>A dataframe containing two columns for study in first column, sample ID (i.e., IID) and in second column, the ancestry label.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_outlier">outlier</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying outlier detection will be performed or not.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_outlierOf">outlierOf</code></td>
<td>
<p>Chracter string, specifying the reference ancestry name for detecting outlier samples. The default is &quot;outlierOf = &quot;EUR&quot;.</p>
</td></tr>
<tr><td><code id="AncestryCheck_:_outlier_threshold">outlier_threshold</code></td>
<td>
<p>Numeric value, specifying the threshold to be be used to detect outlier samples. This threshold will be multiplied with the Eucledean distance from the center of the PC 1 and PC2 to the maximum Euclidean distance of the reference samples. Study samples outside this distance will be considered as outlier. The default is 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with the IDs of non-European samples as outliers. #importFrom DescTools %like%   Removing DescTools from Description imports.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>References</h3>

<p>(1) The International HapMap 3 Consortium. Integrating common and rare genetic variation in diverse human populations. Nature 467, 52–58 (2010). https://doi.org/10.1038/nature09298
(2) The 1000 Genomes Project Consortium. A global reference for human genetic variation. Nature 526, 68–74 (2015). https://doi.org/10.1038/nature15393
(3) Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP, Zondervan KT. Data quality control in genetic case-control association studies. Nat Protoc. 2010 Sep;5(9):1564-73. doi: 10.1038/nprot.2010.116. Epub 2010 Aug 26. PMID: 21085122; PMCID: PMC3025522.
</p>


<h3>Examples</h3>

<pre><code class="language-R">DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
finput &lt;- &quot;GXwasR_example&quot;
reference &lt;- &quot;HapMapIII_NCBI36&quot;
data(&quot;GXwasRData&quot;)
#load(&quot;~/b1137/BBose/GXwasR/data/highLD_hg19.Rda&quot;)
highLD_regions &lt;- highLD_hg19 #&quot;high-LD-regions-hg19-GRCh37.txt&quot; # four columns, chr, start, end, index
#load(&quot;~/b1137/BBose/GXwasR/data/example_data_study_sample_ancestry.Rda&quot;)
study_pop &lt;- example_data_study_sample_ancestry #PreimputeEX
studyLD_window_size = 50
studyLD_step_size = 5
studyLD_r2_threshold = 0.02
filterSNP = TRUE
studyLD = TRUE
referLD = TRUE
referLD_window_size = 50
referLD_step_size = 5
referLD_r2_threshold = 0.02
outlier = TRUE
outlier_threshold = 3
x &lt;- AncestryCheck(DataDir = DataDir, ResultDir = ResultDir, finput = finput, reference = &quot;HapMapIII_NCBI36&quot;,highLD_regions = highLD_regions,
study_pop = study_pop, studyLD = studyLD, referLD = referLD, outlierOf = &quot;EUR&quot;, outlier = outlier, outlier_threshold = outlier_threshold )

#x &lt;- AncestryCheck(DataDir = DataDir, ResultDir = ResultDir, finput = finput, reference = &quot;ThousandGenome&quot;,highLD_regions = highLD_regions,
#study_pop = study_pop, studyLD = studyLD, referLD = referLD, outlierOf = &quot;EUR&quot;, outlier = outlier, outlier_threshold = outlier_threshold )
</code></pre>


</div>
</div>
</div>
<div id="function-4" class="section level1">
<h1>Function 4</h1>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">help</span>(SexCheck, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>SexCheck</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="SexCheck">SexCheck: Compare sex assignments in the input plink files with those imputed from X chromosome inbreeding coefficients</h2>

<h3>Description</h3>

<p>This function compares sex assignments in the input dataset with those predicted from X chromosome inbreeding coefficients following Purcell et. al.,(2007), and gives the option to convert the sex assignments to the predicted values. Implicitly, this function computes observed and expected autosomal homozygous genotype counts for each sample and reports method-of-moments F coefficient estimates (i.e., observed hom. count - expected count) / (total observations - expected count)). The expected counts will be based on loaded or imputed minor allele frequencies.  Since imputed MAFs are highly inaccurate when there are few samples, the &#39;compute freq&#39; parameter should be set to TRUE to compute MAF implicitly.
Due to the use of allele frequencies, if a cohort is comprised of individuals of different ancestries, users may need to process any samples with rare ancestry individually if the dataset has a very unbalanced ancestry distribution. It is advised to run this function with all the parameters set to zero, then examine the distribution of the F estimates (there should be a clear gap between a very tight male clump on the right side of the distribution and the females everywhere else). Then, rerun the function with the parameters that correspond to this gap.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SexCheck(
  DataDir,
  ResultDir = tempdir(),
  finput,
  impute_sex = FALSE,
  compute_freq = FALSE,
  LD = TRUE,
  LD_window_size = 50,
  LD_step_size = 5,
  LD_r2_threshold = 0.02,
  fmax_F = 0.2,
  mmin_F = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SexCheck_:_DataDir">DataDir</code></td>
<td>
<p>Character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="SexCheck_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="SexCheck_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files. Note: Input dataset should contain X and Y regions.</p>
</td></tr>
<tr><td><code id="SexCheck_:_impute_sex">impute_sex</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying sex to be imputed or not. If &#39;TRUE&#39; then sex-imputed PLINK files, prefixed, &#39;seximputed_plink&#39;, will be produced in DataDir.</p>
</td></tr>
<tr><td><code id="SexCheck_:_compute_freq">compute_freq</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying minor allele frequency (MAF). This function requires reasonable MAF estimates, so it is essential to use compute_freq = TRUE for computing MAF from an input PLINK file if there are very few samples in the input dataset. The default is FALSE.</p>
</td></tr>
<tr><td><code id="SexCheck_:_LD">LD</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for applying linkage disequilibrium (LD)-based filtering. The default is TRUE.</p>
</td></tr>
<tr><td><code id="SexCheck_:_LD_window_size">LD_window_size</code></td>
<td>
<p>Integer value, specifying a window size in variant count for LD-based filtering. The default is 50.</p>
</td></tr>
<tr><td><code id="SexCheck_:_LD_step_size">LD_step_size</code></td>
<td>
<p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering. The default is 5.</p>
</td></tr>
<tr><td><code id="SexCheck_:_LD_r2_threshold">LD_r2_threshold</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering. The default is 0.02.</p>
</td></tr>
<tr><td><code id="SexCheck_:_fmax_F">fmax_F</code></td>
<td>
<p>Numeric value between 0 to 1. Samples with F estimates smaller than this value will be labeled as females. The default is 0.2.</p>
</td></tr>
<tr><td><code id="SexCheck_:_mmin_F">mmin_F</code></td>
<td>
<p>Numeric value between 0 to 1. Samples with F estimates larger than this value will be labeled as males. The default is 0.8.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with six columns:FID(Family ID), IID(Individual ID), PEDSEX(Sex as determined in pedigree file (1=male, 2=female)), SNPSEX(Sex as determined by X chromosome), STATUS(Displays &quot;PROBLEM&quot; or &quot;OK&quot; for each individual) and F(The actual X chromosome inbreeding (homozygosity) estimate).A PROBLEM arises if the two sexes do not match, or if the SNP data or pedigree data are ambiguous with regard to sex.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>References</h3>

<p>(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-5" class="section level1">
<h1>Function 5</h1>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">help</span>(Xhwe, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>Xhwe</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="Xhwe">Xhwe: Filter X-chromosome variants for HWE in females.</h2>

<h3>Description</h3>

<p>This function is a part of the post-imputation quality control process prior to GWAS. This tests for Hardy-Weinberg Equilibrium (HWE) for X-chromosome variants in females. Males&#39; hemizygous X chromosome prevents testing for HWE on their haploid X calls, and testing for HWE across all samples would have a high failure rate. This function will check for HWE across the X in females (cases and controls combined), following the recommendation in Khramtsova et al., 2023, and can remove these regions from analysis in all samples. The p-value threshold for filtering out SNPs is 0.05/no.of. X-chromosome variants.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Xhwe(DataDir, ResultDir = tempdir(), finput, filterSNP = TRUE, foutput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Xhwe_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="Xhwe_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="Xhwe_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="Xhwe_:_filterSNP">filterSNP</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for filtering out the X-chromosome variants i.e., SNPs from the input file or not (i.e., only flagged). The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="Xhwe_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files if filtering option for the SNPs is chosen. The default is &quot;FALSE&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing SNPs. If filterSNP = TRUE, the output filtered PLINK binary files will be produced inside DataDir.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-6" class="section level1">
<h1>Function 6</h1>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">help</span>(MAFdiffSexControl, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>MAFdiffSexControl</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="MAFdiffSexControl">MAFdiffSexControl: Test for significantly different minor allele frequency (MAF) between sexes in control samples</h2>

<h3>Description</h3>

<p>With parameters to filter out SNPs and/or flag the SNPs, this function tests for significantly different MAF (p-value &lt; 0.05/no. of SNPs) between sexes in control samples solely for binary phenotypes. Since the disparities may be caused by technical confounding or sample biases for the research cohorts, it is advised that any SNPs in the controls with a sex difference in MAF be carefully evaluated and identified for further examination (Khramtsova et. al., 2023). In autosomal allele frequencies, sex differences are not anticipated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MAFdiffSexControl(
  DataDir,
  ResultDir = tempdir(),
  finput,
  filterSNP = FALSE,
  foutput = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAFdiffSexControl_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="MAFdiffSexControl_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="MAFdiffSexControl_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files with both male and female samples. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="MAFdiffSexControl_:_filterSNP">filterSNP</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for filtering out the SNPs or not (i.e., only flagged). The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="MAFdiffSexControl_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files if filtering option for the SNPs is chosen. The default is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing excluded or flagged SNPs. If filterSNP = TRUE, the output filtered PLINK binary files will be produced inside DataDir.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-7" class="section level1">
<h1>Function 7</h1>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="fu">help</span>(FilterRegion, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>FilterRegion</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="FilterRegion">FilterRegion: Filter chromosomal regions.</h2>

<h3>Description</h3>

<p>Filtering Pseudo-Autosomal Region (PAR), X-transposed region (XTR), Ampliconic, filter based on chromosome code or user-defined regions from input plink files.Only one type of filtering can be done from three types, either by region (using regionfile = TRUE), by chromosome (filterCHR) or by any combination of these three, filterPAR, XTR and Ampliconic.
</p>


<h3>Usage</h3>

<pre><code class="language-R">FilterRegion(
  DataDir,
  ResultDir,
  finput,
  foutput,
  CHRX = TRUE,
  CHRY = FALSE,
  filterPAR = TRUE,
  filterXTR = TRUE,
  filterAmpliconic = TRUE,
  regionfile = FALSE,
  filterCHR = NULL,
  Hg = &quot;19&quot;,
  exclude = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterRegion_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="FilterRegion_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files if the filtering option for the SNPs is chosen. The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_CHRX">CHRX</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; to filter/flag regions from chromosome X. The default is TRUE. Note: CHRX only in effect if one of PAR, XTR or Ampliconic filter is in effect.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_CHRY">CHRY</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; to filter/flag regions from chromosome X. The default is FALSE. Note: CHRY only in effect if one of PAR, XTR or Ampliconic filter is in effect.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_filterPAR">filterPAR</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; to filter out PARs from input plink file. The default is TRUE.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_filterXTR">filterXTR</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; to filter out XTRs from input plink file. The default is TRUE.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_filterAmpliconic">filterAmpliconic</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; to filter out Ampliconic regions from input plink file. The default is TRUE.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_regionfile">regionfile</code></td>
<td>
<p>Character string, specifying the name of the .txt file containing the user-defined regions to be filtered out from input plink file in bed format. The default is FALSE. If regionfile = TRUE, only this filtering will be in effect. Also, PAR, XTR and Ampliconic SNPs from X-chomosome will be flagged and returned.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_filterCHR">filterCHR</code></td>
<td>
<p>Vector value with positive integer, specifying the chromosome code to filter/flag the SNPs. The default is 0, means no filtering based on chromosome code. For non-zero values of this argument, the function will only consider the chromosome code to filter or flag. All other filtering will not work. If filterCHR = TRUE, only this filtering will be in effect. Also, PAR, XTR and Ampliconic SNPs from X-chomosome will be flagged and returned.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_Hg">Hg</code></td>
<td>
<p>Character value, &#39;19&#39;, or &#39;38&#39;, specifying which genome build to use for PAR, XTR and Ampliconic regions. The default is Hg = &quot;19&quot;.</p>
</td></tr>
<tr><td><code id="FilterRegion_:_exclude">exclude</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; to filter and flag or only flag the SNPs. The default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three dataframes: PAR containing SNPs from PAR regions; XTR containing SNPs from XTR region and Ampliconic containing SNPs from Ampliconic region.
</p>
<p>For non-zero value of filterCHR, a dataframe containing the excluded/flagged SNPs will be returned.
</p>
<p>For exclude = TRUE, two sets of plink binary files will be produced in ResultDir. One set will have the remaining SNPs after filtering and other one will have the discarded SNPs.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-8" class="section level1">
<h1>Function 8</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">help</span>(GXwas, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>GXwas</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="GXwas">GXwas: Running genome-wide association study (GWAS) and X-chromosome-wide association study (XWAS) models.</h2>

<h3>Description</h3>

<p>This function runs GWAS models in autosomes with several alternative XWAS models, such as &quot;FMcombx01&quot;,&quot;FMcombx02&quot;,and &quot;FMstatrified&quot; can be applied to both binary and quantitative traits while &quot;GWAcxci&quot; can only be applied to a binary trait. For binary and quantitative features, this function uses logistic and linear regression, allowing for multiple covariates and the interactions with those covariates in a multiple-regression approach. These models are all run using the additive effects of SNPs, and each additional minor allele&#39;s influence is represented by the direction of the regression coefficient. This function attempts to identify the multi-collinearity among predictors by displaying NA for the test statistic and a p-value for all terms in the model. The more terms you add, the more likely you are to run into issues. For details about the different XWAS model, please follow the associated publication.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GXwas(
  DataDir,
  ResultDir,
  finput,
  trait = c(&quot;binary&quot;, &quot;quantitative&quot;),
  standard_beta = TRUE,
  xmodel = c(&quot;FMcombx01&quot;, &quot;FMcombx02&quot;, &quot;FMstatrified&quot;, &quot;GWAScxci&quot;),
  sex = FALSE,
  xsex = FALSE,
  covarfile = NULL,
  interaction = FALSE,
  covartest = c(&quot;ALL&quot;),
  Inphenocov = c(&quot;ALL&quot;),
  combtest = c(&quot;fisher.method&quot;, &quot;fisher.method.perm&quot;, &quot;stouffer.method&quot;),
  MF.zero.sub = 1e-05,
  B = 10000,
  MF.mc.cores = 1,
  MF.na.rm = FALSE,
  MF.p.corr = &quot;none&quot;,
  plot.jpeg = FALSE,
  plotname = &quot;GXwas.plot&quot;,
  snp_pval = 1e-08,
  annotateTopSnp = FALSE,
  suggestiveline = 5,
  genomewideline = 7.3,
  ncores = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GXwas_:_DataDir">DataDir</code></td>
<td>
<p>Character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="GXwas_:_ResultDir">ResultDir</code></td>
<td>
<p>Character string for the folder path where the outputs will be saved.</p>
</td></tr>
<tr><td><code id="GXwas_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files with both male and female samples. This file needs to be in DataDir. Note: Case/control phenotypes are expected to be encoded as 1=unaffected (control), 2=affected (case); 0 is accepted as an alternate missing value encoding. The missing case/control or quantitative phenotypes are expected to be encoded as &#39;NA&#39;/&#39;nan&#39; (any capitalization) or -9.</p>
</td></tr>
<tr><td><code id="GXwas_:_trait">trait</code></td>
<td>
<p>Boolean value, &#39;binary&#39; or &#39;quantitative&#39; for the phenotype i.e. the trait.</p>
</td></tr>
<tr><td><code id="GXwas_:_standard_beta">standard_beta</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; in case of quantitative trait for standardizing the trait or phenotype values (mean 0, unit variance), so the resulting coefficients will be standardized. The default is TRUE.</p>
</td></tr>
<tr><td><code id="GXwas_:_xmodel">xmodel</code></td>
<td>
<p>Models &quot;FMcombx01&quot;,&quot;FMcombx02&quot;,and &quot;FMstatrified&quot; can be chosen for both binary and quantitative traits while &quot;GWAcxci&quot; can only apply to the binary trait. These models take care of the X-chromosomal marker Three female genotypes are coded by 0, 1, and 2 in FM01 and FM02. The two genotypes of males that follow the X-chromosome inactivation (XCI) pattern as random (XCI-R) in the FM01 model are coded by 0 and 1, while the two genotypes that follow the XCI is escaped (XCI-E) in the FM02 model are coded by 0 and 1. To reflect the dose compensation connection between the sexes, FM02 treats men as homozygous females. In the FM01comb and FM01comb methods, associations are tested separately for males and females with the FM01 and FM02 models, respectively, and then the combined p values are computed the Fisher&#39;s method, Fisher&#39;s method with permutation, or Stouffer&#39;s method(1,3-7]. An X-chromosome inactivation (XCI) pattern, or coding technique for X-chromosomal genotypes between sexes, is not required for the XCGA. By simultaneously accounting for four distinct XCI patterns, namely XCI-R, XCI-E, XCI-SN (XCI fully toward normal allele), and XCI-SR (XCI fully toward risk allele), this model may maintain a respectably high power (2). Note that, sex shouldn&#39;t be provided as a covariate in the XCGA model.</p>
</td></tr>
<tr><td><code id="GXwas_:_sex">sex</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for using sex as covariate in association test. It is applicable genome-wide. The default is FALSE.</p>
</td></tr>
<tr><td><code id="GXwas_:_xsex">xsex</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for using sex as covariate in association test for X-chromosomal SNPs. The default is FALSE. This will overwrite &#39;sex&#39; argument for X-chromosome.</p>
</td></tr>
<tr><td><code id="GXwas_:_covarfile">covarfile</code></td>
<td>
<p>Character string for the full name of the covariate file in .txt format. This file should be placed in DataDir. Note about the covariate file: The first column of this file should be FID, the second column should be IID and the other columns should be covariates. The primary header line should be there starting with “FID”, and “IID” followed by covariate names. If an individual is not present in the covariate file, or if the individual has a missing phenotype value (i.e. -9 by default) for the covariate, then that individual is set to missing (i.e. will be excluded from association analysis). It is important to note that for statrified GWAS model, if PCs are included as covar then it should be generated separately for each cohort and then included in the covarfile. Use the function &quot;DummyCovar()&quot; to generate a new covariate file with categorical variables down-coded as binary dummy variables for the covariate file with categorical variables. For instance, if a variable has K categories, K-1 new dummy variables are constructed, and the original covariate is now estimated with a coefficient for each category.</p>
</td></tr>
<tr><td><code id="GXwas_:_interaction">interaction</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for including SNP x covariate interaction term/terms from the association analysis. The default is FALSE. If a permutation procedure is chosen then the interaction will be automatically FALSE. For the interaction with the two covariates COV1 and COV2, the model will look like: Y = b0 + b1.ADD + b2.COV1 + b3.COV2 + b4.ADD x COV1 + b5.ADD x COV2 + e. When interaction factors are incorporated into the model, the main effects&#39; significance is not always determined simply; rather, it depends on the arbitrary coding of the variables. To put it another way, you should probably just interpret the p-value for the interaction. Also, The p-values for the covariates do not represent the test for the SNP-phenotype association after controlling for the covariate. That is the first row (ADD). Rather, the covariate term is the test associated with the covariate-phenotype association. These p-values might be extremely significant (e.g. if one covaries for smoking in an analysis of heart disease, etc) but this does not mean that the SNP has a highly significant effect necessarily. Note that, this feature is not valid for XCGA model for XWAS part</p>
</td></tr>
<tr><td><code id="GXwas_:_covartest">covartest</code></td>
<td>
<p>Vector value with &quot;NULL&quot;,&quot;ALL&quot; or covarite name/names to be included in the test. The default is NULL. For instance, the user can choose “AGE” and “SEX” as covartest = c(“AGE”, “SEX”) or all the covariates as covartest = c(“ALL”).</p>
</td></tr>
<tr><td><code id="GXwas_:_Inphenocov">Inphenocov</code></td>
<td>
<p>Vector of integer values starting from 1 to extract the terms which user wants from the above model: Y = b0 + b1.ADD + b2.COV1 + b3.COV2 + b4.ADDxCOV1 + b5.ADDxCOV2 + e. The terms will appear in order as (1) for ADD, (2) for COV1, (4) for ADD x COV1, and (5) for ADD x COV2. If the user wants to extract the terms for COV1 and ADD x COV1, she needs to specify it as c(2,4). The default is c(“ALL”). Note that, this feature is not valid for the XCGA model for the XWAS part.</p>
</td></tr>
<tr><td><code id="GXwas_:_combtest">combtest</code></td>
<td>
<p>Character vector specifying method for combining p-values for stratified GWAS with FM01comb and FM02comb XWAS models. Choices are “stouffer.method”, &quot;fisher.method&quot; and &quot;fisher.method.perm&quot;. For fisher.method the function for combining p-values uses a statistic, S = -2 ∑^k /log p, which follows a χ^2 distribution with 2k degrees of freedom (3).
For fisher.method.perm, using p-values from stratified tests, the summary statistic for combining p-values is S = -2 ∑ /log p. A p-value for this statistic can be derived by randomly generating summary statistics (4). Therefore, a p-value is randomly sampled from each contributing study, and a random statistic is calculated. The fraction of random statistics greater or equal to S then gives the final p-value.
</p>
<p>For stouffer.method ,the function applies Stouffer’s method (6]) to the p-values assuming that the p-values to be combined are independent. Letting p1, p2, . . . , pk denote the individual (one- or two-sided) p-values of the k hypothesis tests to be combined, the test statistic is then computed with $z = ∑^k_1fracz_isqrt(k)$ where $z_i$ = Φ−1 (1 – $p_i$) and Φ −1 (·) denotes the inverse of the cumulative distribution function of a standard normal distribution. Under the joint null hypothesis, the test statistic follows a standard normal distribution which is used to compute the combined p-value. This functionality is taken from the R package poolr (7).
Note that only p-values between 0 and 1 are allowed to be passed to these methods.</p>
</td></tr>
<tr><td><code id="GXwas_:_MF.zero.sub">MF.zero.sub</code></td>
<td>
<p>Small numeric value for substituting p-values of 0 in in stratified GWAS with FM01comb and FM02comb XWAS models. The default is 0.00001. As log(0) results in Inf this replaces p-values of 0 by default with a small float.</p>
</td></tr>
<tr><td><code id="GXwas_:_B">B</code></td>
<td>
<p>Integer value specifying the number of permutation in case of using fisher.method.perm method in stratified GWAS with FM01comb and FM02comb XWAS models. The default is 10000.</p>
</td></tr>
<tr><td><code id="GXwas_:_MF.mc.cores">MF.mc.cores</code></td>
<td>
<p>Number of cores used for fisher.method.perm in stratified GWAS with FM01comb and FM02comb XWAS models.</p>
</td></tr>
<tr><td><code id="GXwas_:_MF.na.rm">MF.na.rm</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for removing p-values of NA in stratified GWAS with FM01comb and FM02comb XWAS in case of using Fisher’s and Stouffer’s methods. The default is FALSE.</p>
</td></tr>
<tr><td><code id="GXwas_:_MF.p.corr">MF.p.corr</code></td>
<td>
<p>Character vector specifying method for correcting the summary p-values for FMfcomb and FMscomb models. Choices are &quot;bonferroni&quot;, &quot;BH&quot; and &quot;none&quot; for Bonferroni,  Benjamini-Hochberg and none, respectively. The default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="GXwas_:_plot.jpeg">plot.jpeg</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for saving the plots in .jpeg file. The default is TRUE.</p>
</td></tr>
<tr><td><code id="GXwas_:_plotname">plotname</code></td>
<td>
<p>A character string specifying the prefix of the file for plots. This file will be saved in DataDir. The default is &quot;GXwas.plot&quot;.</p>
</td></tr>
<tr><td><code id="GXwas_:_snp_pval">snp_pval</code></td>
<td>
<p>Numeric value as p-value threshold for annotation. SNPs below this p-value will be annotated on the plot. The default is 1e-08.</p>
</td></tr>
<tr><td><code id="GXwas_:_annotateTopSnp">annotateTopSnp</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE. If TRUE, it only annotates the top hit on each chromosome that is below the snp_pval threshold. The default is FALSE.</p>
</td></tr>
<tr><td><code id="GXwas_:_suggestiveline">suggestiveline</code></td>
<td>
<p>The default is 5 (for p-value 1e-05).</p>
</td></tr>
<tr><td><code id="GXwas_:_genomewideline">genomewideline</code></td>
<td>
<p>The default is 7.3 (for p-value 5e-08).</p>
</td></tr>
<tr><td><code id="GXwas_:_ncores">ncores</code></td>
<td>
<p>Integer value, specifying the number of cores for parallel processing. The default is 0 (no parallel computation).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with GWAS (with XWAS for X-chromosomal variants) along with Manhattan and Q-Q plots.
In the case of the stratified test, the return is a list containing three dataframes, namely, FWAS, MWAS, and MFWAS with association results in only female, only male, and both cohorts, respectively. This will be accompanied by Miami and Q-Q plots. The individual manhattan and Q-Q-plots for stratified tests prefixed with xmodel type will be in the DataDir.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>References</h3>

<p>(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.
(2) Su Y, Hu J, Yin P, Jiang H, Chen S, Dai M, Chen Z, Wang P. XCMAX4: A Robust X Chromosomal Genetic Association Test Accounting for Covariates. Genes (Basel). 2022 May 9;13(5):847. doi: 10.3390/genes13050847. PMID: 35627231; PMCID: PMC9141238.
(3) Fisher, R.A. (1925). Statistical Methods for Research Workers. Oliver and Boyd (Edinburgh);
(4) Rhodes, D. R., (2002). Meta-analysis of microarrays: interstudy validation of gene expression profiles reveals pathway dysregulation in prostate cancer. Cancer research, 62(15), 4427-33;
(5) Moreau, Y.et al. (2003). Comparison and meta-analysis of microarray data: from the bench to the computer desk. Trends in Genetics, 19(10), 570-577.
(6) Stouffer, S. A., Suchman, E. A., DeVinney, L. C., Star, S. A., &amp; Williams, R. M., Jr. (1949). The American Soldier: Adjustment During Army Life (Vol. 1). Princeton, NJ: Princeton University Press.
(7) Cinar, O. &amp; Viechtbauer, W. (2022). The poolr package for combining independent and dependent p values. Journal of Statistical Software, 101(1), 1–42. https://doi.org/10.18637/jss.v101.i01
</p>


<h3>Examples</h3>

<pre><code class="language-R">
DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir = tempdir()
finput &lt;- &quot;GXwasR_example&quot;
standard_beta = TRUE
xsex = FALSE
sex = TRUE
Inphenocov = NULL
covartest = NULL
interaction = FALSE
MF.na.rm = FALSE
B = 10000
MF.zero.sub = 0.00001
trait = &quot;binary&quot;
xmodel = &quot;FMcombx02&quot;
combtest = &quot;fisher.method&quot;
snp_pval = 1e-08
covarfile = NULL
ncores = 0
MF.mc.cores = 1
ResultGXwas &lt;- GXwas(DataDir = DataDir,ResultDir = ResultDir, finput = finput, xmodel = xmodel, trait = trait, covarfile = covarfile, sex = sex, xsex = xsex, combtest = combtest, MF.p.corr = &quot;none&quot;, snp_pval = snp_pval, plot.jpeg = TRUE, suggestiveline = 5, genomewideline = 7.3,MF.mc.cores = 1, ncores = ncores)
</code></pre>


</div>
</div>
</div>
<div id="function-9" class="section level1">
<h1>Function 9</h1>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">help</span>(PvalComb, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>PvalComb</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="PvalComb">Title: This function combines the p-values of two separate GWAS summary statistics (for instance male and female populations), merges them, and then applies various statistical methods (like Stouffer&#39;s method, Fisher&#39;s method) to integrate the p-values. It also includes functionality for generating plots like Manhattan plots and Q-Q plots.</h2>

<h3>Description</h3>

<p>Title: This function combines the p-values of two separate GWAS summary statistics (for instance male and female populations), merges them, and then applies various statistical methods (like Stouffer&#39;s method, Fisher&#39;s method) to integrate the p-values. It also includes functionality for generating plots like Manhattan plots and Q-Q plots.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PvalComb(
  SumstatMale,
  SumstatFemale,
  combtest,
  MF.p.corr = &quot;none&quot;,
  MF.zero.sub = 1e-05,
  MF.na.rm = TRUE,
  MF.mc.cores = 1,
  B = 1000,
  plot.jpeg = TRUE,
  plotname = &quot;GXwas.plot&quot;,
  PlotDir = tempdir(),
  snp_pval,
  annotateTopSnp = FALSE,
  suggestiveline = 5,
  genomewideline = 7.3,
  ncores = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PvalComb_:_SumstatMale">SumstatMale</code></td>
<td>
<p>R dataframe object of summary statistics of male GWAS with five mandatory columns: &quot;CHR&quot; (numeric chromosome code),&quot;SNP&quot; (variant id), &quot;A1&quot; (allele) ,&quot;POS&quot; (base-pair position) and &quot;P&quot; (p-value). Other coulmns may present.</p>
</td></tr>
<tr><td><code id="PvalComb_:_SumstatFemale">SumstatFemale</code></td>
<td>
<p>R dataframe object of summary statistics of female GWAS with five mandatory columns: &quot;SNP&quot;, &quot;A1&quot;, &quot;TEST&quot; , &quot;POS&quot; and &quot;P&quot;. Other coulmns may present.</p>
</td></tr>
<tr><td><code id="PvalComb_:_combtest">combtest</code></td>
<td>
<p>Character vector specifying method for combining p-values for stratified GWAS models. Choices are “stouffer.method”, &quot;fisher.method&quot; and &quot;fisher.method.perm&quot;. For fisher.method the function for combining p-values uses a statistic, S = -2 ∑^k /log p, which follows a χ^2 distribution with 2k degrees of freedom (3).
For fisher.method.perm, using p-values from stratified tests, the summary statistic for combining p-values is S = -2 ∑ /log p. A p-value for this statistic can be derived by randomly generating summary statistics (4). Therefore, a p-value is randomly sampled from each contributing study, and a random statistic is calculated. The fraction of random statistics greater or equal to S then gives the final p-value.</p>
</td></tr>
<tr><td><code id="PvalComb_:_MF.p.corr">MF.p.corr</code></td>
<td>
<p>Character vector specifying method for correcting the summary p-values for FMfcomb and FMscomb models. Choices are &quot;bonferroni&quot;, &quot;BH&quot; and &quot;none&quot; for Bonferroni,  Benjamini-Hochberg and none, respectively. The default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="PvalComb_:_MF.zero.sub">MF.zero.sub</code></td>
<td>
<p>Small numeric value for substituting p-values of 0 in GWAS summary statistics. The default is 0.00001. As log(0) results in Inf this replaces p-values of 0 by default with a small float.</p>
</td></tr>
<tr><td><code id="PvalComb_:_MF.na.rm">MF.na.rm</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for removing p-values of NA in stratified GWAS summary satistics in case of using Fisher’s and Stouffer’s methods. The default is TRUE.</p>
</td></tr>
<tr><td><code id="PvalComb_:_MF.mc.cores">MF.mc.cores</code></td>
<td>
<p>Number of cores used for fisher.method.perm for combining p-values. The default is 1.</p>
</td></tr>
<tr><td><code id="PvalComb_:_B">B</code></td>
<td>
<p>Integer value specifying the number of permutation in case of using fisher.method.perm method. The default is 10000.</p>
</td></tr>
<tr><td><code id="PvalComb_:_plot.jpeg">plot.jpeg</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for saving the plots in .jpeg file. The default is TRUE.</p>
</td></tr>
<tr><td><code id="PvalComb_:_plotname">plotname</code></td>
<td>
<p>A character string specifying the prefix of the file for plots. This file will be saved in DataDir. The default is &quot;GXwas.plot&quot;.</p>
</td></tr>
<tr><td><code id="PvalComb_:_PlotDir">PlotDir</code></td>
<td>
<p>A character string specifying the path of the directory where the plots will be saved. The default is tempdir().</p>
</td></tr>
<tr><td><code id="PvalComb_:_snp_pval">snp_pval</code></td>
<td>
<p>Numeric value as p-value threshold for annotation. SNPs below this p-value will be annotated on the plot. The default is 1e-08.</p>
</td></tr>
<tr><td><code id="PvalComb_:_annotateTopSnp">annotateTopSnp</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE. If TRUE, it only annotates the top hit on each chromosome that is below the snp_pval threshold. The default is FALSE.</p>
</td></tr>
<tr><td><code id="PvalComb_:_suggestiveline">suggestiveline</code></td>
<td>
<p>Numeric value for suggestive cut-off line in GWAS manhattan plot. The default is 5 (for p-value 1e-05).</p>
</td></tr>
<tr><td><code id="PvalComb_:_genomewideline">genomewideline</code></td>
<td>
<p>Numeric value for genome-wide significant cut-off line in GWAS manhattan plot. The default is 7.3 (for p-value 5e-08).</p>
</td></tr>
<tr><td><code id="PvalComb_:_ncores">ncores</code></td>
<td>
<p>Integer value, specifying the number of cores for parallel processing. The default is 0 (no parallel computation).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe with GWAS summary statistics (with XWAS for X-chromosomal variants) along with Manhattan and Q-Q plots.
</p>


<h3>References</h3>

<p>(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.
(2) Su Y, Hu J, Yin P, Jiang H, Chen S, Dai M, Chen Z, Wang P. XCMAX4: A Robust X Chromosomal Genetic Association Test Accounting for Covariates. Genes (Basel). 2022 May 9;13(5):847. doi: 10.3390/genes13050847. PMID: 35627231; PMCID: PMC9141238.
(3) Fisher, R.A. (1925). Statistical Methods for Research Workers. Oliver and Boyd (Edinburgh);
(4) Rhodes, D. R., (2002). Meta-analysis of microarrays: interstudy validation of gene expression profiles reveals pathway dysregulation in prostate cancer. Cancer research, 62(15), 4427-33;
(5) Moreau, Y.et al. (2003). Comparison and meta-analysis of microarray data: from the bench to the computer desk. Trends in Genetics, 19(10), 570-577.
(6) Stouffer, S. A., Suchman, E. A., DeVinney, L. C., Star, S. A., &amp; Williams, R. M., Jr. (1949). The American Soldier: Adjustment During Army Life (Vol. 1). Princeton, NJ: Princeton University Press.
(7) Cinar, O. &amp; Viechtbauer, W. (2022). The poolr package for combining independent and dependent p values. Journal of Statistical Software, 101(1), 1–42. https://doi.org/10.18637/jss.v101.i01
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(&quot;GXwasRData&quot;)
SumstatMale &lt;- Mfile
colnames(SumstatMale)[3]&lt;- &quot;POS&quot;
SumstatFemale &lt;- Ffile
colnames(SumstatFemale)[3]&lt;- &quot;POS&quot;
# Use the below datasets for tutorial document
# SumstatMale &lt;- Summary_Stat_Ex1
# colnames(SumstatMale)[3]&lt;- &quot;POS&quot;
# SumstatFemale &lt;- Summary_Stat_Ex2
# colnames(SumstatFemale)[3]&lt;- &quot;POS&quot;
PvalComb_Result &lt;- PvalComb(SumstatMale = SumstatMale, SumstatFemale = SumstatFemale, combtest = &quot;fisher.method&quot;, MF.mc.cores = 1, snp_pval = 0.001, plot.jpeg = FALSE, suggestiveline = 3, genomewideline = 5.69897, ncores = 1)

</code></pre>


</div>
</div>
</div>
<div id="function-10" class="section level1">
<h1>Function 10</h1>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">help</span>(SexDiff, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>SexDiff</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="SexDiff">SexDiff: Sex difference in effect size for each SNP using t-test.</h2>

<h3>Description</h3>

<p>This function uses the GWAS summary statistics from sex-stratified tests like FM01comb or FM02comb, to evaluate the difference in effect size between males and females at each SNP using a t-test.
</p>
<p>The input dataframes should only include X-chromosome in order to obtain results for sex differences based solely on X-linked loci.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SexDiff(Mfile, Ffile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SexDiff_:_Mfile">Mfile</code></td>
<td>
<p>R dataframe of summary statistics of GWAS or XWAS of male samples with six mandatory columns, SNP(Variant),CHR(Chromosome number)
,BP(Base pair position),A1(Minor allele),BETA_M(Effect size) and SE_M(Standard error). This can be generated by running FM01comb or
FM02comb model with GXWAS function.</p>
</td></tr>
<tr><td><code id="SexDiff_:_Ffile">Ffile</code></td>
<td>
<p>R dataframe of summary statistics of GWAS or XWAS of male samples with six mandatory columns, SNP(Variant),CHR(Chromosome number)
,BP(Base pair position),A1(Minor allele),BETA_F(Effect size) and SE_F(Standard error). This can be generated by running FM01comb or
FM02comb model with GXWAS function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>R dataframe with seven columns, SNP(Variant),CHR(Chromosome number)
,BP(Base pair position),A1(Minor allele), tstat(t-statistics for effect-size test), P(p-value) and adjP (Bonferroni corrected p-value).
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(&quot;GXwasRData&quot;)
Difftest &lt;- SexDiff(Mfile,Ffile)
significant_snps &lt;- Difftest[Difftest$adjP &lt;0.05,]# 9
</code></pre>


</div>
</div>
</div>
<div id="function-11" class="section level1">
<h1>Function 11</h1>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">help</span>(SexDiffZscore, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<pre><code>## No documentation for &#39;SexDiffZscore&#39; in specified packages and libraries</code></pre>
</div>
<div id="function-12" class="section level1">
<h1>Function 12</h1>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">help</span>(DiffZeroOne, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>DiffZeroOne</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="DiffZeroOne">DiffZeroOne: Assessing the Z-score for deviation from one and zero.</h2>

<h3>Description</h3>

<p>This function tests the null hypothesis that a measured statistics (example: genetic correlation, rg for a trait) &lt; 1 using a 1-tailed test compared with a normal distribution (z = (1 − measure statistics)/Standard error).
For multiple tests, users are encouraged to apply a Bonferroni multiple-testing correction.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DiffZeroOne(inputdata, diffzero = TRUE, diffone = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiffZeroOne_:_inputdata">inputdata</code></td>
<td>
<p>A dataframe object, contaning three columns, namely, &#39;Trait&#39; (i.e., the phenotype of interest), &#39;Stat&#39; (i.e., the measured statistics) and &#39;SE&#39; (i.e., the standard error of the measured statistics).</p>
</td></tr>
<tr><td><code id="DiffZeroOne_:_diffzero">diffzero</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying to perform diviation from 0 test.</p>
</td></tr>
<tr><td><code id="DiffZeroOne_:_diffone">diffone</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying to perform diviation from 1 test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with columns, &#39;Trait&#39;,&#39;Stat&#39;,&#39;SE&#39;, &#39;P0&#39; (i.e, p-value for deviation from zero test) and &#39;P1&#39; (i.e., p-value for deviation from 1 test)
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(&quot;GXwasRData&quot;)
colnames(Example_rgdata) &lt;- c(&quot;Trait&quot;,&quot;Stat&quot;,&quot;SE&quot;)
inputdata = Example_rgdata
x &lt;- DiffZeroOne(inputdata,FALSE,TRUE)
</code></pre>


</div>
</div>
</div>
<div id="function-13" class="section level1">
<h1>Function 13</h1>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">help</span>(TestXGene, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>TestXGene</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="TestXGene">TestXGene: Performing gene-based association test using GWAS/XWAS summary statistics.</h2>

<h3>Description</h3>

<p>This function performs gene-based association tests using GWAS/XWAS summary statistics and SNP-SNP correlation matrices. For  SNP-SNP correlation matrices, users have the flexibility to use either the base genotype data or 1000 Genomes Phase 3 reference genotype data. Users also have options to define the regional positions of genes to include the SNPs according to their investigation. This function computes gene-wise SNP-SNP correlation matrices and can perform nine different gene-based tests, such as, “BT&quot; (burden test), &quot;SKAT&quot; (sequence kernel association test), &quot;SKATO&quot; (combination of BT and SKAT), &quot;sumchi&quot; (sum of χ2-statistics), &quot;ACAT&quot; (aggregated Cauchy association test for combining P values), &quot;PCA&quot;(principal component approach), &quot;FLM&quot;( functional multiple linear regression model), &quot;simpleM&quot; (Bonferroni correction test), &quot;minp&quot; (minimum P-value) leveraging PLINK1.9 (1) and sumFREGAT (2,3) tools.
</p>
<p>Though this function implicitly performs X-linked gene-based test, it is flexible to perform this analysis genome-wide. For the details about the different tests, please follow the associated paper.
</p>


<h3>Usage</h3>

<pre><code class="language-R">TestXGene(
  DataDir,
  ResultDir = tempdir(),
  finput,
  sumstat,
  gene_file,
  gene_range = 5e+05,
  score_file,
  ref_data = NULL,
  max_gene = NULL,
  sample_size = NULL,
  genebasedTest = c(&quot;SKAT&quot;, &quot;SKATO&quot;, &quot;sumchi&quot;, &quot;ACAT&quot;, &quot;BT&quot;, &quot;PCA&quot;, &quot;FLM&quot;, &quot;simpleM&quot;,
    &quot;minp&quot;),
  gene_approximation = TRUE,
  beta_par,
  weights_function,
  geno_variance_weights,
  kernel_p_method = &quot;kuonen&quot;,
  acc_devies = 1e-08,
  lim_devies = 1e+06,
  rho = TRUE,
  skato_p_threshold = 0.8,
  anno_type = &quot;&quot;,
  mac_threshold,
  reference_matrix_used,
  regularize_fun,
  pca_var_fraction = 0.85,
  flm_basis_function = &quot;fourier&quot;,
  flm_num_basis = 25,
  flm_poly_order = 4,
  flip_genotypes = FALSE,
  omit_linear_variant = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TestXGene_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the all the input files.</p>
</td></tr>
<tr><td><code id="TestXGene_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="TestXGene_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files for the genotype data. This file is used to compute the correlation between the SNPs. This file needs to be in DataDir. If the base genotype data is unavailable, then users can use the 1000 Genomes Project samples. Users should use the population that most closely represents the base sample. For ACAT model, this parameter is not mandatory and could be set NULL.</p>
</td></tr>
<tr><td><code id="TestXGene_:_sumstat">sumstat</code></td>
<td>
<p>A dataframe object with GWAS summary statistics. When the base-genotype data is used to compute genetic correlations, the mandatory columns are Column 1: &quot;CHROM&quot; (i.e., chromosome code), Column 2: &quot;POS&quot; (i.e., base-pair position), Column 3: &quot;ID&quot; (i.e. SNP IDs), Column 4: “P” (i.e., p-values), Column 5: “BETA” (i.e., effect-size), Column 6: &quot;A1&quot; (i.e., effect allele), Column 7: “A2” (i.e., alternative allele) and Column 8: &quot;EAF&quot; (i.e., the effect allele frequency) are mandatory when base-genotype data is used to compute genetic correlations. Otherwise, if the users are using reference data, then columns 5 to 8 are optional. Also, in that case, columns, such as &quot;REF&quot; (i.e., reference allele), and &quot;ALT&quot; (i.e., alternative allele) could be present to compare alleles with those in the reference file and exclude genetic variants if alleles do not match. There could be an additional column, “ANNO” with functional annotations (like &quot;intron_variant&quot;, &quot;synonymous&quot;, &quot;missense&quot; etc.)</p>
</td></tr>
<tr><td><code id="TestXGene_:_gene_file">gene_file</code></td>
<td>
<p>Character string, specifying the prefix of the name of a .txt file listing genes in refFlat format. This file needs to be in DataDir. The X-linked gene files, &quot;Xlinkedgenes_hg19.txt&quot; and &quot;Xlinkedgenes_hg38.txt&quot; and autosomal gene files, “Autosomes_hg19.txt” and “Autosomes_hg38.txt” can be specified. The default is &quot;Xlinkedgenes_hg19.txt&quot;. The genome built should be in agreement with the analysis.</p>
</td></tr>
<tr><td><code id="TestXGene_:_gene_range">gene_range</code></td>
<td>
<p>Integer value, specifying the up stream and down stream range (in kilo base) of a gene for SNPs to be considered. The default is 500000.</p>
</td></tr>
<tr><td><code id="TestXGene_:_score_file">score_file</code></td>
<td>
<p>Character string, specifying the prefix of a file which will be used to produce score files with Z scores from P values and beta input from GWAS summary statistics.</p>
</td></tr>
<tr><td><code id="TestXGene_:_ref_data">ref_data</code></td>
<td>
<p>Character string, specifying the path to a reference dataframe with additional data needed to recode user data according to correlation matrices that will be used. It contains &quot;ID&quot; column with names of  SNPs,  &quot;REF&quot; and &quot;ALT&quot; columns with alleles that were coded as 0 and 1, respectively. Effect sizes from data will be inverted for variants with effect alleles different from &quot;ALT&quot; alleles in reference data. If presented, &quot;REF&quot; and &quot;ALT&quot; columns from the input data will be used to sort out variants with alleles different from those in reference data. This dataframe  can also be a source of map data and allele frequencies if they are not present in data. &quot;AF&quot; column in the reference file represents the allele frequency of &quot;ALT&quot; allele. The default is &quot;ref1KG.MAC5.EUR_AF.RData&quot;.</p>
</td></tr>
<tr><td><code id="TestXGene_:_max_gene">max_gene</code></td>
<td>
<p>Positive integer value, specifying the number of genes for which the gene-based test will be performed. The default is NULL to consider all the genes.</p>
</td></tr>
<tr><td><code id="TestXGene_:_sample_size">sample_size</code></td>
<td>
<p>Positive integer value, specifying the sample size of the GWAS. Only needed for FLM and PCA models.</p>
</td></tr>
<tr><td><code id="TestXGene_:_genebasedTest">genebasedTest</code></td>
<td>
<p>Character string, specifying the name of the gene-based test. Nine different tests can be specified, &quot;SKAT&quot;,&quot;SKATO&quot;,&quot;sumchi&quot;,&quot;ACAT&quot;,&quot;BT&quot;,&quot;PCA&quot;,&quot;FLM&quot;,&quot;simpleM&quot;,&quot;minp&quot;. The default is &quot;SKAT&quot;.</p>
</td></tr>
<tr><td><code id="TestXGene_:_gene_approximation">gene_approximation</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether approximation for large genes (&gt;= 500 SNPs) should be used. Applicable for SKAT, SKATO, sumchi, PCA, FLM. The default is TRUE for these methods).</p>
</td></tr>
<tr><td><code id="TestXGene_:_beta_par">beta_par</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether approximation for large genes (&gt;= 500 SNPs) should be used. Applicable for SKAT, SKATO, sumchi, PCA, FLM (default = TRUE for these methods).</p>
</td></tr>
<tr><td><code id="TestXGene_:_weights_function">weights_function</code></td>
<td>
<p>A function of MAF to assign weights for each genetic variant. By default is NULL. In this case the weights will be calculated using the beta distribution.</p>
</td></tr>
<tr><td><code id="TestXGene_:_geno_variance_weights">geno_variance_weights</code></td>
<td>
<p>Character string, indicating whether scores should be weighted by the variance of genotypes: &quot;none&quot; (i.e., no weights applied, resulting in a sum chi-square test); &quot;se.beta&quot; (i.e., scores weighted by variance of genotypes estimated from P values and effect sizes); &quot;af&quot; (i.e., scores weighted by variance of genotypes calculated as AF * (1 - AF), where AF is allele frequency.</p>
</td></tr>
<tr><td><code id="TestXGene_:_kernel_p_method">kernel_p_method</code></td>
<td>
<p>Character string, specifying the method for computing P value in kernel-based tests, such as SKAT, SKATO and sumchi. Available methods are &quot;kuonen&quot; (9), &quot;davies&quot; (10) and &quot;hybrid&quot; (11). The default is &quot;kuonen&quot;.</p>
</td></tr>
<tr><td><code id="TestXGene_:_acc_devies">acc_devies</code></td>
<td>
<p>Positive numeric value, specifying the accuracy parameter for &quot;davies&quot; method. The default is 1e-8.</p>
</td></tr>
<tr><td><code id="TestXGene_:_lim_devies">lim_devies</code></td>
<td>
<p>Positive numeric value, specifying the limit parameter for &quot;davies&quot; method. The default is 1e+6.</p>
</td></tr>
<tr><td><code id="TestXGene_:_rho">rho</code></td>
<td>
<p>Logical value, &#39;TRUE&#39; or &#39;FALSE&#39; or can be a vector of grid values from 0 to 1. If TRUE, the optimal test (SKAT-O) is performed (12). The default grid is c(0, 0.1^2, 0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1).</p>
</td></tr>
<tr><td><code id="TestXGene_:_skato_p_threshold">skato_p_threshold</code></td>
<td>
<p>Positive numeric value, specifying the largest P value that will be considered as important when performing computational optimization in SKAT-O. All P values larger than skato_p_threshold will be processed via burden test. The default is 0.8</p>
</td></tr>
<tr><td><code id="TestXGene_:_anno_type">anno_type</code></td>
<td>
<p>A character (or character vector) indicating annotation types to be used. The default is &quot;&quot; (i.e, nothing).</p>
</td></tr>
<tr><td><code id="TestXGene_:_mac_threshold">mac_threshold</code></td>
<td>
<p>Integer value, specifying the threshold of MACs (Minor allele content) calculated from MAFs. In ACAT, scores with MAC &lt;= 10 will be combined using Burden test.</p>
</td></tr>
<tr><td><code id="TestXGene_:_reference_matrix_used">reference_matrix_used</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or&#39;FALSE&#39; logical indicating whether the correlation matrices were generated using the reference matrix. The default is FALSE. If  TRUE, regularization algorithms will be applied to ensure the invertibility and numerical stability of the matrices.</p>
</td></tr>
<tr><td><code id="TestXGene_:_regularize_fun">regularize_fun</code></td>
<td>
<p>Character string, specifying the one of two regularization algorithms if ‘reference_matrix’ is TRUE:  &#39;LH&#39; (default) or &#39;derivLH&#39;. Currently, both give similar results.</p>
</td></tr>
<tr><td><code id="TestXGene_:_pca_var_fraction">pca_var_fraction</code></td>
<td>
<p>Positive numeric value, specifying the minimal proportion of genetic variance within the region that should be explained by principal components used in PCA method. This is also valid in &#39;simpleM&#39;. The default is 0.85.</p>
</td></tr>
<tr><td><code id="TestXGene_:_flm_basis_function">flm_basis_function</code></td>
<td>
<p>Character string, specifying the name of a basis function type for beta-smooth in FLM method. Can be set to &quot;bspline&quot; (B-spline basis) or &quot;fourier&quot; (Fourier basis, default).</p>
</td></tr>
<tr><td><code id="TestXGene_:_flm_num_basis">flm_num_basis</code></td>
<td>
<p>Positive integer value, specifying the number of basis functions to be used for beta-smooth in FLM method. The default is 25.</p>
</td></tr>
<tr><td><code id="TestXGene_:_flm_poly_order">flm_poly_order</code></td>
<td>
<p>Positive integer value, specifying the polynomial order to be used in &quot;bspline&quot; for FLM model. The default = 4, which corresponds to the cubic B-splines. This has no effect if only Fourier bases are used</p>
</td></tr>
<tr><td><code id="TestXGene_:_flip_genotypes">flip_genotypes</code></td>
<td>
<p>Logical value, &#39;TRUE&#39; or &#39;FALSE&#39;, indicating whether the genotypes of some genetic variants should be flipped (relabelled) for their better functional representation (13). The default is FALSE.</p>
</td></tr>
<tr><td><code id="TestXGene_:_omit_linear_variant">omit_linear_variant</code></td>
<td>
<p>Logical value, &#39;TRUE&#39; or &#39;FALSE&#39;, indicating whether to omit linearly dependent genetic variants. It was done in the FLM test (4). The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns &quot;gene&quot;,  &quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;markers&quot; (i.e., numbers of SNPs), &quot;filtered.markers&quot; (i.e. filtered SNPs) and &quot;pvalue&quot; (i.e., p-value). Additionally, for  “BT”, there will be “beta” (i.e., gene-level estimates of betas) and “beta.se” (i.e., standard errors of betas).  For “FLM”, there will be the “model” column with the names of the functional models used for each region. Names shortly describe the functional basis and the number of basis functions used. E.g., &quot;F25&quot; means 25 Fourier basis functions, &quot;B15&quot; means 15 B-spline basis functions. For “PCA”, there will be the “ncomponents” (the number of the principal components used for each region) and “explained.variance.fraction”(i.e., the proportion of genetic variance they make up) columns.
</p>


<h3>References</h3>

<p>(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.
(2) Svishcheva GR, Belonogova NM, Zorkoltseva IV, Kirichenko AV, Axenovich TI. Gene-based association tests using GWAS summary statistics. Bioinformatics. 2019 Oct 1;35(19):3701-3708. doi: 10.1093/bioinformatics/btz172. PMID: 30860568.
(3) Belonogova NM, Svishcheva GR, Kirichenko AV, Zorkoltseva IV, Tsepilov YA, Axenovich TI. sumSTAAR: A flexible framework for gene-based association studies using GWAS summary statistics. PLoS Comput Biol. 2022 Jun 2;18(6):e1010172. doi: 10.1371/journal.pcbi.1010172. PMID: 35653402; PMCID: PMC9197066.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-14" class="section level1">
<h1>Function 14</h1>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="fu">help</span>(MetaGXwas, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<pre><code>## No documentation for &#39;MetaGXwas&#39; in specified packages and libraries</code></pre>
</div>
<div id="function-15" class="section level1">
<h1>Function 15</h1>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">help</span>(ComputePRS, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>ComputePRS</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="ComputePRS">ComputePRS: Computing polygenic risk score (PRS)</h2>

<h3>Description</h3>

<p>This function calculates the polygenic risk score, which is the total of allele counts (genotypes) weighted by estimated effect sizes from genome-wide association studies. It uses C+T filtering techniques. The users could perform clumping procedure choromosome-wise and genome-wide. Also, the function offers the choice of including several genetic principal components along with other covariates. Using this function, users have the freedom to experiment with various clumping and thresholding arrangements to test a wide range of various parameter values.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ComputePRS(
  DataDir,
  ResultDir = tempdir(),
  finput,
  summarystat,
  phenofile,
  covarfile = NULL,
  effectsize = c(&quot;BETA&quot;, &quot;OR&quot;),
  ldclump = FALSE,
  LDreference,
  clump_p1,
  clump_p2,
  clump_r2,
  clump_kb,
  byCHR = TRUE,
  pthreshold = c(0.001, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5),
  highLD_regions,
  ld_prunning = FALSE,
  window_size = 50,
  step_size = 5,
  r2_threshold = 0.02,
  nPC = 6,
  pheno_type = &quot;binary&quot;
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePRS_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the all the input files.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="ComputePRS_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files for the genotype data i.e., the target data based on which clumping procedure will be performed. This file needs to be in DataDir. If your target data are small (e.g. N &lt; 500) then you can use the 1000 Genomes Project samples. Make sure to use the population that most closely reflects represents the base sample.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_summarystat">summarystat</code></td>
<td>
<p>A dataframe object with GWAS summary statistics. The mandatory column headers in this dataframe are &#39;CHR&#39;(Chromosome code), &#39;BP&#39;(Basepair position), &#39;A1&#39; (effect allele), ‘SNP’ (i.e., SNP idenitifier), ‘BETA’ or &quot;OR&quot;(i.e., effect-size or logarithm of odds ratio), and ‘P’ (i.e., p-values). Special Notes: The first three columns needed to be &quot;SNP&quot;, &quot;A1&quot; and &quot;BETA&quot; or &quot;OR&quot;.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_phenofile">phenofile</code></td>
<td>
<p>A character string, specifying the name of the mandatory phenotype file. This is a plain text file with no header line; columns family ID, individual ID and phenotype columns. For binary trait, the phenotypic value should be coded as 0 or 1, then it will be recognized as a case-control study (0 for controls and 1 for cases). Missing value should be represented by &quot;-9&quot; or &quot;NA&quot;. The interested phenotype column should be labeled as &quot;Pheno1&quot;. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_covarfile">covarfile</code></td>
<td>
<p>A character string, specifying the name of the covariate file which is a plain .text file with no header line; columns are family ID, individual ID and the covariates. The default is NULL. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_effectsize">effectsize</code></td>
<td>
<p>Boolean value, ‘BETA’ or ‘OR’, specifying the type of the GWAS effectsize. The default is ‘BETA’.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_ldclump">ldclump</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to perform clumping or not.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_LDreference">LDreference</code></td>
<td>
<p>A character string, specifying the  prefix of the PLINK files of the population reference panel of the same ancestry, and ideally the one that was used for imputing your target dataset. These files should be in DataDir.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_clump_p1">clump_p1</code></td>
<td>
<p>Numeric value, specifying the significance threshold for index SNPs if &#39;ldclump&#39; was set to be TRUE. The default is 0.0001.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_clump_p2">clump_p2</code></td>
<td>
<p>Numeric value, specifying the secondary significance threshold for clumped SNPs if &#39;ldclump&#39; was set to be TRUE. The default is 0.01</p>
</td></tr>
<tr><td><code id="ComputePRS_:_clump_r2">clump_r2</code></td>
<td>
<p>Numeric value, specifying the linkage disequilibrium (LD) threshold for clumping if &#39;ldclump&#39; was set to be TRUE. The default is 0.50.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_clump_kb">clump_kb</code></td>
<td>
<p>Integer value, specifying the physical distance threshold in base-pair for clumping if &#39;ldclump&#39; was set to be TRUE. The default is 250.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_byCHR">byCHR</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying chromosome-wise clumping procedure if &#39;ldclump&#39; was set to be TRUE. The default is TRUE.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_pthreshold">pthreshold</code></td>
<td>
<p>Numeric vector, containing several p value thresholds to maximize predictive ability of the derived polygenic scores.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_highLD_regions">highLD_regions</code></td>
<td>
<p>Character string, specifying the .txt file name with known genomic regions with high LD. The default is NULL.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_ld_prunning">ld_prunning</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39; for LD-based filtering for computing genetic PC as covariates.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_window_size">window_size</code></td>
<td>
<p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering in computing genetic PC. The default is 50.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_step_size">step_size</code></td>
<td>
<p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering in computing genetic PCs. The default is 5.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_r2_threshold">r2_threshold</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering in computing genetic PCs. The default is 0.02.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_nPC">nPC</code></td>
<td>
<p>Positive integer value, specifying the number of genetic PCs to be included as predictor in the PRS model fit. The default is 6.</p>
</td></tr>
<tr><td><code id="ComputePRS_:_pheno_type">pheno_type</code></td>
<td>
<p>Boolean value, ‘binary’ or ‘quantitative’, specifying the type of the trait. The default is ‘binary’.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing a dataframe and a numeric value. The dataframe,PRS, contains four mandatory columns, such as,
IID (i.e., Individual ID), FID (i.e., Family ID), Pheno1 (i.e., the trait for PRS) and Score (i.e., the best PRS). Other columns of covariates could be there. The numeric value, BestP contains the threshold of
of the best p-value for the best pRS model fit.
</p>
<p>Also, the function produces several plots such as p-value thresholds vs PRS model fit and PRS distribution among male and females. For case-control data, it shows PRS distribution among cases and controls and ROC curves as well.
</p>


<h3>Examples</h3>

<pre><code class="language-R">DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
finput &lt;- &quot;GXwasR_example&quot;
data(&quot;GXwasRData&quot;)
summarystat &lt;- Summary_Stat_Ex1[,c(2,4,7,1,3,12)]
phenofile &lt;- Example_phenofile #Cannot be NULL, the interested phenotype column should be labeled as &quot;Pheno1&quot;.
covarfile &lt;- Example_covarfile
clump_p1 = 0.0001
clump_p2 = 0.0001
clump_kb = 500
clump_r2 = 0.5
byCHR = TRUE
pthreshold &lt;- Example_pthresoldfile$Threshold
ld_prunning &lt;- TRUE
highLD_regions &lt;- highLD_hg19
window_size &lt;- 50
step_size &lt;- 5
r2_threshold &lt;- 0.02
nPC = 6 #We can incorporate PCs into our PRS analysis to account for population stratification.
pheno_type = &quot;binary&quot;

PRSresult &lt;- ComputePRS(DataDir,ResultDir,finput,summarystat,phenofile,covarfile,effectsize=&quot;BETA&quot;,LDreference = &quot;GXwasR_example&quot;, ldclump = FALSE,clump_p1,clump_p2,clump_r2,clump_kb, byCHR = TRUE,pthreshold = pthreshold,highLD_regions = highLD_regions, ld_prunning = TRUE,window_size = 50, step_size = 5,r2_threshold = 0.02, nPC = 6,pheno_type = &quot;binary&quot;)

## This table shows 10 samples with phenotype, covariates and a PRS column.
PRS &lt;- PRSresult$PRS
PRS[1:10,]
## The best threshold
BestPvalue &lt;- PRSresult$BestP$Threshold
BestPvalue
</code></pre>


</div>
</div>
</div>
<div id="function-16" class="section level1">
<h1>Function 16</h1>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">help</span>(ComputeCorrBT, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<pre><code>## No documentation for &#39;ComputeCorrBT&#39; in specified packages and libraries</code></pre>
</div>
<div id="function-17" class="section level1">
<h1>Function 17</h1>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">help</span>(EstimateHerit, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>EstimateHerit</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="EstimateHerit">EstimateHerit: Computing SNP heritability i.e., the proportion of phenotypic variance explained by SNPs.</h2>

<h3>Description</h3>

<p>This functions performs two types of heritability estimation, (i)GREML:Genomic relatedness matrix (GRM) restricted maximum likelihood-based method following GCTA (1) and (ii)LDSC: LD score regression-based method following (2,3). For the details, please follow the associated paper.
</p>
<p>Prior using this function, users are recommended to apply QCsnp and QCsample to ensure data quality control.
</p>


<h3>Usage</h3>

<pre><code class="language-R">EstimateHerit(
  DataDir,
  ResultDir,
  finput,
  summarystat = NULL,
  ncores = 2,
  model = c(&quot;LDSC&quot;, &quot;GREML&quot;),
  computeGRM = TRUE,
  grmfile_name = NULL,
  byCHR = FALSE,
  r2_LD = 0,
  LDSC_blocks = 20,
  REMLalgo = c(0, 1, 2),
  nitr = 100,
  cat_covarfile = NULL,
  quant_covarfile = NULL,
  prevalance = 0.01,
  partGRM = FALSE,
  autosome = TRUE,
  Xsome = TRUE,
  nGRM = 3,
  cripticut = 0.025,
  minMAF = NULL,
  maxMAF = NULL,
  hg = c(&quot;hg19&quot;, &quot;hg38&quot;),
  PlotIndepSNP = c(TRUE, FALSE),
  IndepSNP_window_size = 50,
  IndepSNP_step_size = 5,
  IndepSNP_r2_threshold = 0.02,
  highLD_regions = NULL,
  plotjpeg = TRUE,
  plotname = &quot;Heritability_Plots&quot;
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EstimateHerit_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the all the input files.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files for the genotype data. This file needs to be in DataDir. For LDSC model, if the original genotype data is not available, Hapmap 3 or 1000Genome data can be used.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_summarystat">summarystat</code></td>
<td>
<p>A dataframe object with GWAS summary statistics. The mandatory column headers in this dataframe are &#39;chr&#39;(Chromosome code), &#39;pos&#39;(Basepair position), &#39;a1&#39; (First allele code), ‘rsid’ (i.e., SNP idenitifier), ‘beta’ (i.e., effect-size or logarithm of odds ratio), ‘beta_se’ (i.e., standard error of beta), ‘P’ (i.e., p-values) and &#39;n_eff&#39; (i.e., effective sample size).  For case-control study, effective sample size should be 4 / (1/&lt;# of cases&gt; + 1/&lt;# of controls&gt;). The default is NULL.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_ncores">ncores</code></td>
<td>
<p>Integer value, specifying the number of cores to be used for running LDSC model. The default is 2.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_model">model</code></td>
<td>
<p>Character string, specifying the heritability estimation model. There are two options, “GREML” or “LDSC”. The default is “GREML”.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_computeGRM">computeGRM</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to compute GRM matrices or not. The default is TRUE.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_grmfile_name">grmfile_name</code></td>
<td>
<p>A string of characters specifying the prefix of autosomal .grm.bin file.
</p>
<p>Users need to provide separate GRM files for autosomes and X chromosome in ResultDir.
</p>
<p>The X chromosomal GRM file should have &quot;x&quot; added in the autosomal prefix as file name.
</p>
<p>For instance, if autosomal file is &quot;ABC.grm.bin&quot;, then X chromosomal file should be &quot;xABC.grm.bim&quot;.
</p>
<p>If you are providing chromosome-wise GRMs, then the prefix should add &quot;ChrNumber_&quot; at the starting
</p>
<p>of the prefix like, &quot;Chr1_ABC.grm.bin&quot;. The default is NULL.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_byCHR">byCHR</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether the analysis will be performed chromosome wise or not. The default is FALSE.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_r2_LD">r2_LD</code></td>
<td>
<p>Numeric value, specifying the LD threshold for clumping in LDSC model. The default is 0.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_LDSC_blocks">LDSC_blocks</code></td>
<td>
<p>Integer value, specifying the block size for performing jackknife variance estimator in LDSC model following (3). The default is 200.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_REMLalgo">REMLalgo</code></td>
<td>
<p>Integer value of 0, 1 or 2, specifying the algorithm to run REML iterations, 0 for average information (AI), 1 for Fisher-scoring and 2 for EM. The default option is 0, i.e. AI-REML (1).</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_nitr">nitr</code></td>
<td>
<p>Integer value, specifying the number of iterations for performing the REML. The default is 100.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_cat_covarfile">cat_covarfile</code></td>
<td>
<p>A character string, specifying the name of the categorical covariate file which is a plain text file with no header line; columns are family ID, individual ID and discrete covariates. The default is NULL. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_quant_covarfile">quant_covarfile</code></td>
<td>
<p>A character string, specifying the name of the quantitative covariate file which is a plain text file with no header line; columns are family ID, individual ID and continuous covariates. The default is NULL. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_prevalance">prevalance</code></td>
<td>
<p>Numeric value, specifying the disease prevalence. The default is 0.01. Note: for the continuous trait value, users should use &#39;prevalence = 0&#39;.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_partGRM">partGRM</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether the GRM will be partitioned into n parts (by row) in GREML model. The default is FALSE.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_autosome">autosome</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether estimate of heritability will be done for autosomes or not. The default is ‘TRUE’.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_Xsome">Xsome</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether estimate of heritability will be done for X chromosome or not. The default is ‘TRUE’.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_nGRM">nGRM</code></td>
<td>
<p>Integer value, specifying the number of the partision of the GRM in GREML model. The default is 3.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_cripticut">cripticut</code></td>
<td>
<p>Numeric value, specifying the threshold to create a new GRM of &quot;unrelated&quot; individuals in GREML model. The default is arbitrary chosen as 0.025 following (1).</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_minMAF">minMAF</code></td>
<td>
<p>Positive numeric value (0,1), specifying the minimum threshold for the MAF filter of the SNPs in the GREML model. This value cannot be greater than &#39;maxMAF&#39; parameter. The default is NULL. For NULL, maximum MAF value of the genotype data will be computed and printed on the plot.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_maxMAF">maxMAF</code></td>
<td>
<p>Positive numeric value (0,1), specifying the maximum threshold for the MAF filter of the SNPs in the GREML model. This value cannot be less than &#39;minMAF&#39; parameter. The default is NULL. For NULL, minimum MAF value of the genotype data will be computed and printed on the plot.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_hg">hg</code></td>
<td>
<p>Boolean value, specifying the genome built, “hg19” or “hg38” to use chromosome length from UCSC genome browser and getting genes and proteins according to this built. The default is “hg19”.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_PlotIndepSNP">PlotIndepSNP</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to use independent SNPs i.e., chromosome-wise LD pruned SNPs in the plots or not. The default is TRUE.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_IndepSNP_window_size">IndepSNP_window_size</code></td>
<td>
<p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering. The default is 50.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_IndepSNP_step_size">IndepSNP_step_size</code></td>
<td>
<p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering for pruned SNPs in the plots. The default is 5.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_IndepSNP_r2_threshold">IndepSNP_r2_threshold</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering for pruned SNPs in the plots. The default is 0.02.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_highLD_regions">highLD_regions</code></td>
<td>
<p>Character string, specifying the .txt file name with genomic regions with high LD for using in finding pruned SNPs in the plots. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_plotjpeg">plotjpeg</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to save the plots in jpeg file in ResultDir. The default is TRUE.</p>
</td></tr>
<tr><td><code id="EstimateHerit_:_plotname">plotname</code></td>
<td>
<p>String of character value specifying the name of the jpeg file with the plots. The deafult is &quot;Heritability_Plots&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with maximum eight columns for GREML (here, three columns if running genome-wide) and ten columns for LDSC model if byCHR is TRUE. The columns, such as, &quot;chromosome&quot;(i.e., chromosome code),&quot;snp_proportion&quot; (i.e.,chromosome-wise SNP propotion)&quot;, &quot;no.of.genes&quot; (i.e., number of genes per chromosome), &quot;no.of.proteins&quot; (i.e., number of genes per chromosome),&quot;size_mb&quot; (i.e., chromosome length), &quot;Source&quot; (i.e., source of heritability), &quot;Variance&quot; (i.e., estimated heritability), and &quot;SE&quot; (i.e., standard error of the estimated heritability) are common for both GREML and LDSC model. The column, &quot;Intercept&quot; (i.e., LDSC regression intercept) and &quot;Int_SE&quot; (i.e., standard error of the intercept) will be two extra columns for LDSC models. Source column will have rows, such as V(1) (i.e., name of genetic variance), V(e) (i.e., residual variance), V(p) (i.e., phenotypic variance), V(1)/Vp (i.e., ratio of genetic variance to phenotypic variance), and V(1)/Vp_L (i.e., ratio of genetic variance to phenotypic variance in liability scale for binary phenotypes). If byCHR is FALSE, then the first five columns will not be reported in the dataframe.
</p>


<h3>References</h3>

<p>(1)  Yang et al. (2011) GCTA: a tool for Genome-wide Complex Trait Analysis. Am J Hum Genet. 88(1): 76-82.
</p>
<p>(2)  Bulik-Sullivan B, et al. LD score regression distinguishes confounding from polygenicity in genome-wide association studies. Nat Genet. 2014;47:291–295.
</p>
<p>(3)  Florian Privé, Julyan Arbel, Bjarni J Vilhjálmsson, LDpred2: better, faster, stronger, Bioinformatics, Volume 36, Issue 22-23, 1 December 2020, Pages 5424–5431, https://doi.org/10.1093/bioinformatics/btaa1029
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Not Run
#DataDir = system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
#ResultDir = tempdir()
#finput &lt;- &quot;GXwasR_example&quot;
#load(paste0(&quot;/projects/b1137/BBose/GXwasR/data/Summary_Stat_Ex1.Rda&quot;))
#data(&quot;GXwasRData&quot;)
#test.sumstats &lt;- na.omit(Summary_Stat_Ex1[Summary_Stat_Ex1$TEST==&quot;ADD&quot;,c(1:4,6:8)])
#colnames(test.sumstats) &lt;- c(&quot;chr&quot;,&quot;rsid&quot;,&quot;pos&quot;,&quot;a1&quot;,&quot;n_eff&quot;,&quot;beta&quot;,&quot;beta_se&quot;)
#summarystat = test.sumstats
#ncores = 3
#model = &quot;GREML&quot;
#byCHR = FALSE
#r2_LD = 0
#LDSC_blocks = 20
#REMLalgo = 0
#nitr = 3
#cat_covarfile = NULL
#quant_covarfile = NULL
#prevalance = 0.01
#partGRM = FALSE
#autosome = TRUE
#Xsome = TRUE
#nGRM = 3
#cripticut = 0.025
#minMAF = NULL
#maxMAF = NULL
#hg = &quot;hg19&quot;
#PlotIndepSNP = TRUE
#IndepSNP_window_size = 50
#IndepSNP_step_size = 5
#IndepSNP_r2_threshold = 0.02
#highLD_regions = &quot;high-LD-regions-hg19-GRCh37.txt&quot;
#H2 &lt;- EstimateHerit(DataDir = DataDir, ResultDir = ResultDir, finput = finput, summarystat = NULL, ncores, model = &quot;GREML&quot;, byCHR = &quot;TRUE&quot;, r2_LD = 0, LDSC_blocks = 20,REMLalgo = 0, nitr = 100, cat_covarfile = NULL, quant_covarfile = NULL,prevalance = 0.01, partGRM = FALSE, autosome = TRUE, Xsome = TRUE, nGRM = 3,cripticut = 0.025, minMAF = NULL, maxMAF = NULL,hg = &quot;hg19&quot;,PlotIndepSNP = TRUE, IndepSNP_window_size = 50,IndepSNP_step_size = 5,IndepSNP_r2_threshold = 0.02,highLD_regions = highLD_hg19)
</code></pre>


</div>
</div>
</div>
<div id="function-18" class="section level1">
<h1>Function 18</h1>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="fu">help</span>(SexRegress, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>SexRegress</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="SexRegress">SexRegress: Performing linear regression analysis with quantitative response variable.</h2>

<h3>Description</h3>

<p>This function could be used to check association of two variables. For instance, PRS with sex.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SexRegress(fdata, regressor_index, response_index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SexRegress_:_fdata">fdata</code></td>
<td>
<p>R dataframe object. The column with header &quot;response&quot; should contain the response variable. All other column are the regressor.</p>
</td></tr>
<tr><td><code id="SexRegress_:_regressor_index">regressor_index</code></td>
<td>
<p>Integer value, specifying the column number of the main regressor variable.</p>
</td></tr>
<tr><td><code id="SexRegress_:_response_index">response_index</code></td>
<td>
<p>Integer value, specifying the column number of the response variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value containing the regression estimate (&quot;Estimate&quot;), standard error (&quot;Std. Error&quot;), statistics (&quot;t value&quot;) and p-value (&quot;Pr(&gt;|t|)&quot;)
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(&quot;GXwasRData&quot;)
fdata = Regression_Ex
fdata$SEX &lt;- as.factor(as.character(fdata$SEX))
response_index &lt;- 1
regressor_index &lt;- 2

x &lt;- SexRegress(fdata, regressor_index, response_index)

</code></pre>


</div>
</div>
</div>
<div id="function-19" class="section level1">
<h1>Function 19</h1>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="fu">help</span>(FilterPlinkSample, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>FilterPlinkSample</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="FilterPlinkSample">FilterPlinkSample: Making PLINK files with desired samples.</h2>

<h3>Description</h3>

<p>This function prepares PLINK binary files with the desired samples.
</p>


<h3>Usage</h3>

<pre><code class="language-R">FilterPlinkSample(
  DataDir,
  ResultDir,
  finput,
  foutput = NULL,
  filter_sample = &quot;cases&quot;,
  keep_remove_sample_file = NULL,
  keep = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterPlinkSample_:_DataDir">DataDir</code></td>
<td>
<p>Character string for the file path of the all input files.</p>
</td></tr>
<tr><td><code id="FilterPlinkSample_:_ResultDir">ResultDir</code></td>
<td>
<p>character string for the file path where the output PLINK files will be stored.</p>
</td></tr>
<tr><td><code id="FilterPlinkSample_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterPlinkSample_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterPlinkSample_:_filter_sample">filter_sample</code></td>
<td>
<p>Character string, specifying the sample type to be retained. The choices are, &quot;cases&quot;,&quot;controls&quot;,&quot;males&quot; and &quot;females&quot;. The default is &quot;cases&quot;.</p>
</td></tr>
<tr><td><code id="FilterPlinkSample_:_keep_remove_sample_file">keep_remove_sample_file</code></td>
<td>
<p>Character string, specifying the prefix of a space/tab-delimited text file with no header. For the samples that we want to keep or remove, the family IDs should be in the first column and within-family IDs in the second column. This file needs to be in the DataDir. The default is NULL.</p>
</td></tr>
<tr><td><code id="FilterPlinkSample_:_keep">keep</code></td>
<td>
<p>Boolean value, &quot;TRUE&quot; or &quot;FALSE&quot; for specifying desired samples to keep or remove. The default is &quot;TRUE&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
The output plink files with passed samples will be saved in ResultDir.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-20" class="section level1">
<h1>Function 20</h1>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="fu">help</span>(ComputeGeneticPC, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>ComputeGeneticPC</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="ComputeGeneticPC">ComputeGeneticPC: Computing principal components from genetic relationship matrix</h2>

<h3>Description</h3>

<p>This function performs principal components analysis (PCA) based on the variance-standardized relationship matrix (1).
Top principal components are generally used as covariates in association analysis regressions to help correct for population stratification
</p>


<h3>Usage</h3>

<pre><code class="language-R">ComputeGeneticPC(
  DataDir,
  ResultDir = tempdir(),
  finput,
  countPC = 10,
  plotPC = TRUE,
  highLD_regions = NULL,
  ld_prunning = TRUE,
  window_size = 50,
  step_size = 5,
  r2_threshold = 0.02
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputeGeneticPC_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_countPC">countPC</code></td>
<td>
<p>Integer value, specifying the number of principal components. The default is 10.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_plotPC">plotPC</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to plot the first two PCs.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_highLD_regions">highLD_regions</code></td>
<td>
<p>A R dataframe with genomic regions with high LD for using in finding pruned SNPs in the plots. The default is NULL.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_ld_prunning">ld_prunning</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering for pruned SNPs in the plots. The default is 0.02.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_window_size">window_size</code></td>
<td>
<p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering. The default is 50.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_step_size">step_size</code></td>
<td>
<p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering for pruned SNPs in the plots. The default is 5.</p>
</td></tr>
<tr><td><code id="ComputeGeneticPC_:_r2_threshold">r2_threshold</code></td>
<td>
<p>Numeric value between 0 to 1 of pairwise r^2 threshold for LD-based filtering for pruned SNPs in the plots. The default is 0.02.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with genetic principal components. The first two columns are IID (i.e., Individual Id) and FID (i.e., Family ID). The other columns are PCs.
</p>


<h3>References</h3>

<p>(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#&#39; # Not Run
DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
finput &lt;- &quot;GXwasR_example&quot;
data(&quot;GXwasRData&quot;)
#highLD_regions &lt;- &quot;high-LD-regions-hg19-GRCh37.txt&quot;
highLD_regions &lt;- highLD_hg19
ld_prunning &lt;- &quot;TRUE&quot;
window_size &lt;- 50
step_size &lt;- 5
r2_threshold &lt;- 0.02
countPC &lt;- 20
# GP &lt;- ComputeGeneticPC(DataDir = DataDir,ResultDir=ResultDir,finput=finput,highLD_regions = highLD_hg19, countPC = 20)
</code></pre>


</div>
</div>
</div>
<div id="function-21" class="section level1">
<h1>Function 21</h1>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="fu">help</span>(ClumpLD, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>ClumpLD</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="ClumpLD">ClumpLD: Clumping SNPs using linkage disequilibrium between SNPs</h2>

<h3>Description</h3>

<p>This function, which is based on empirical estimations of linkage disequilibrium between SNPs, groups the SNP-based results across one or more datasets or analysis. This approach can be used in two basic scenarios: (i) To summarize the top X single SNP findings from a genome-wide scan as fewer clusters of connected SNPs (i.e., to assess how many independent loci are associated). (ii) To give researchers a simple approach to merge sets of data from multiple studies when those studies may have used various marker sets for genotyping.
The clumping process begins with the index SNPs that are significant at threshold p1 and have not yet been clumped. It then creates clumps of all additional SNPs that are within a specified kb of the index SNP and that are in linkage disequilibrium with the index SNP based on an r-squared threshold. Following that, these SNPs are filtered based on the outcome for that SNP. As this method is greedy (1), each SNP will, at most, only appear in one clump. The P value and ALLELES would always, at random, be chosen from the first input file if the same SNP appeared in several input files in SNPdata argument. Instead of the best p-value, the function refer to the SNP that has the strongest LD to the index as the best proxy. Based on the genotype data, the SNP with the highest LD will be the same for all input files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ClumpLD(
  DataDir,
  finput,
  SNPdata,
  ResultDir = tempdir(),
  clump_p1,
  clump_p2,
  clump_r2,
  clump_kb,
  byCHR = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClumpLD_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="ClumpLD_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files which will be used to calculate linkage disequilibrium between the SNPs. This actual genotype data may or may not be the same dataset that was used to generate the summary statistics. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="ClumpLD_:_SNPdata">SNPdata</code></td>
<td>
<p>a list of R dataframes containing a single or multiple summary statistics with SNP and P (i.e., p-values) in mandatory column headers. Other columns could be present.</p>
</td></tr>
<tr><td><code id="ClumpLD_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="ClumpLD_:_clump_p1">clump_p1</code></td>
<td>
<p>Numeric value, specifying the significance threshold for index SNPs. The default is 0.0001.</p>
</td></tr>
<tr><td><code id="ClumpLD_:_clump_p2">clump_p2</code></td>
<td>
<p>Numeric value, specifying the secondary significance threshold for clumped SNPs. The default is 0.01</p>
</td></tr>
<tr><td><code id="ClumpLD_:_clump_r2">clump_r2</code></td>
<td>
<p>Numeric value, specifying the LD threshold for clumping. The default is 0.50.</p>
</td></tr>
<tr><td><code id="ClumpLD_:_clump_kb">clump_kb</code></td>
<td>
<p>Integer value, specifying the physical distance threshold in base-pair for clumping. The default is 250.</p>
</td></tr>
<tr><td><code id="ClumpLD_:_byCHR">byCHR</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying whether to perform the clumping chromosome-wise.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with twelve columns:&#39;CHR&#39;(Chromosome code), &#39;F&#39;(SNPdata fileset code as 1,2,...), &#39;SNP&#39;(SNP identifier), &#39;BP&#39;(Physical position of SNP inbase-pairs),&#39;TOTAL&#39;(Total number of other SNPs in clump i.e. passing clump_kb and clump_r2 thresholds), &#39;NSIG&#39;(Number of clumped SNPs that are not significant p &gt; 0.05), &#39;S05&#39;(Number of clumped SNPs 0.01 &lt; p &lt; 0.05), &#39;S01&#39;(Number of clumped SNPs 0.001 &lt; p &lt; 0.01), &#39;S001&#39;(Number of clumped SNPs 0.0001 &lt; p &lt; 0.001), &#39;S0001&#39;(Number of clumped SNPs p &lt; 0.0001), &#39;SP2&#39;(List of SNPs names and SNPdata code) clumped and significant at clump_p2 threshold.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>References</h3>

<p>(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.
</p>


<h3>Examples</h3>

<pre><code class="language-R">DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir = tempdir()
finput &lt;- &quot;GXwasR_example&quot;
data(&quot;GXwasRData&quot;)
#load(paste0(&quot;/projects/b1137/BBose/GXwasR/data/Summary_Stat_Ex1.Rda&quot;))
#load(paste0(&quot;/projects/b1137/BBose/GXwasR/data/Summary_Stat_Ex2.Rda&quot;))
SNPdata &lt;- list(Summary_Stat_Ex1,Summary_Stat_Ex2)
clump_p1 = 0.0001
clump_p2 = 0.001
clump_r2 = 0.5
clump_kb = 250
byCHR = TRUE
clumpedResult &lt;- ClumpLD(DataDir,finput,SNPdata,ResultDir,clump_p1,clump_p2,clump_r2,clump_kb, byCHR)
</code></pre>


</div>
</div>
</div>
<div id="function-22" class="section level1">
<h1>Function 22</h1>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="fu">help</span>(GetMFPlink, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>GetMFPlink</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="GetMFPlink">GetMFPlink: Getting male and female PLINK binary files.</h2>

<h3>Description</h3>

<p>This function prepares separate male and female PLINK binary files from combined PLINK files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GetMFPlink(
  DataDir,
  ResultDir = tempdir(),
  finput,
  foutput,
  sex,
  xplink = FALSE,
  autoplink = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetMFPlink_:_DataDir">DataDir</code></td>
<td>
<p>Character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="GetMFPlink_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="GetMFPlink_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="GetMFPlink_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files.</p>
</td></tr>
<tr><td><code id="GetMFPlink_:_sex">sex</code></td>
<td>
<p>Boolean value, &#39;males&#39; or &#39;females&#39;, specifying output plink binary files with male or female samples.</p>
</td></tr>
<tr><td><code id="GetMFPlink_:_xplink">xplink</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying output plink binary files with only X chromosome or not. Default is FALSE.</p>
</td></tr>
<tr><td><code id="GetMFPlink_:_autoplink">autoplink</code></td>
<td>
<p>Boolean value, &#39;TRUE&#39; or &#39;FALSE&#39;, specifying output plink binary files with only autosome or not. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-23" class="section level1">
<h1>Function 23</h1>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">help</span>(plinkVCF, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>plinkVCF</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="plinkVCF">plinkVCF: Converting VCF files to plink binary files and vice-versa.</h2>

<h3>Description</h3>

<p>This function performs the conversion between VCF files to plink binary formats.
</p>
<p>For VCF to plink files conversion, if you do not specify any FAM file when you are converting from VCF to plink format, then plink will just create a &#39;dummy&#39; FAM file with the same name as your dataset with missing phenotypes and missing sex.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plinkVCF(
  DataDir,
  ResultDir = tempdir(),
  finput,
  foutput,
  VtoP = FALSE,
  PtoV = TRUE,
  Famfile = NULL,
  PVbyCHR = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plinkVCF_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input plink binary files and all other input files.</p>
</td></tr>
<tr><td><code id="plinkVCF_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="plinkVCF_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input plink binary files or vcf files. This file needs to be in DataDir.</p>
</td></tr>
<tr><td><code id="plinkVCF_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output plink binary files if filtering option for the SNPs is chosen. The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="plinkVCF_:_VtoP">VtoP</code></td>
<td>
<p>Boolean value, TRUE or FALSE, specifying the conversion of VCF files to plink binary files or not. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plinkVCF_:_PtoV">PtoV</code></td>
<td>
<p>Boolean value, TRUE or FALSE, specifying the conversion of plink binary files to VCF  files or not. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plinkVCF_:_Famfile">Famfile</code></td>
<td>
<p>Character string, specifying the name of the original .fam file if VtoP was set to be TRUE. This file needs to be in DataDir. The default is NULL.</p>
</td></tr>
<tr><td><code id="plinkVCF_:_PVbyCHR">PVbyCHR</code></td>
<td>
<p>Boolean value, TRUE or FALSE specifying to do the plink to vcf conversion chromosome-wise or not. The default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
</p>
<p>The output files will be saved in ResultDir.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
# finput &lt;- &quot;GXwasR_example&quot; #Plink file
# foutput &lt;- &quot;GXwasR_example1&quot;
# DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
# ResultDir &lt;- tempdir()
# PtoV = TRUE
# VtoP = FALSE
# Famfile = NULL
# PVbyCHR = FALSE
# x &lt;- plinkVCF(DataDir, ResultDir, finput, foutput, VtoP, PtoV, Famfile, PVbyCHR)

</code></pre>


</div>
</div>
</div>
<div id="function-24" class="section level1">
<h1>Function 24</h1>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">help</span>(MergeRegion, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>MergeRegion</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="MergeRegion">MergeRegion: Merging two sets of plink binary files.</h2>

<h3>Description</h3>

<p>This function combines the two genotype datasets based on either common SNPs or all the SNPs between them.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MergeRegion(
  DataDir,
  ResultDir,
  finput1,
  finput2,
  foutput,
  use_common_snps = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MergeRegion_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input plink binary files.</p>
</td></tr>
<tr><td><code id="MergeRegion_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="MergeRegion_:_finput1">finput1</code></td>
<td>
<p>Character string, specifying the prefix of the first input PLINK binary files.</p>
</td></tr>
<tr><td><code id="MergeRegion_:_finput2">finput2</code></td>
<td>
<p>Character string, specifying the prefix of the first input PLINK binary files.</p>
</td></tr>
<tr><td><code id="MergeRegion_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files if filtering option for the SNPs is chosen. The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="MergeRegion_:_use_common_snps">use_common_snps</code></td>
<td>
<p>Boolean value, TRUE or FALSE, specifying to use common SNPs for merging or to use all the SNPs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
</p>
<p>The output plink files will be saved in ResultDir.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Not Run
# DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
# ResultDir &lt;- tempdir()
# finput1 &lt;- &quot;GXwasR_example&quot;
# finput2 &lt;- &quot;GXwasR_example_imputed&quot;
# foutput &lt;- &quot;Test_output&quot;
# Not Run
#y &lt;- MergeRegion(DataDir, ResultDir, finput1, finput2, foutput,  use_common_snps = TRUE)
</code></pre>


</div>
</div>
</div>
<div id="function-25" class="section level1">
<h1>Function 25</h1>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">help</span>(FilterAllele, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>FilterAllele</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="FilterAllele">FilterAllele: Filtering out the multi-allelic variants</h2>

<h3>Description</h3>

<p>This function filters out the multi-allelic SNPs from the input dataset.
</p>


<h3>Usage</h3>

<pre><code class="language-R">FilterAllele(DataDir, ResultDir, finput, foutput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterAllele_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterAllele_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="FilterAllele_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterAllele_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files. If multi-allelic variants are present, this file will be produced after filtering out these variants.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
After multi-allelic variant filtering, the filtered plink files with only biallelic SNPs will be saved in ResultDir.
</p>


<h3>Author(s)</h3>

<p>Banabithi Bose
</p>


<h3>Examples</h3>

<pre><code class="language-R">DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
finput &lt;- &quot;GXwasR_example&quot;
foutput &lt;- &quot;Filter_Test&quot;
x &lt;- FilterAllele(DataDir, ResultDir, finput, foutput)
</code></pre>


</div>
</div>
</div>
<div id="function-26" class="section level1">
<h1>Function 26</h1>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">help</span>(PlinkSummary, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>PlinkSummary</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="PlinkSummary">PlinkSummary: Summary of plink format genotype dataset</h2>

<h3>Description</h3>

<p>PlinkSummary: Summary of plink format genotype dataset
</p>


<h3>Usage</h3>

<pre><code class="language-R">PlinkSummary(DataDir, ResultDir = tempdir(), finput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlinkSummary_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="PlinkSummary_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path of the plink program to be set up.</p>
</td></tr>
<tr><td><code id="PlinkSummary_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files. This file needs to be in DataDir.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class="language-R">DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
finput &lt;- &quot;GXwasR_example&quot;

x &lt;- PlinkSummary(DataDir,ResultDir, finput)

</code></pre>


</div>
</div>
</div>
<div id="function-27" class="section level1">
<h1>Function 27</h1>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="fu">help</span>(FilterSNP, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>FilterSNP</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="FilterSNP">Title: Filter out SNPs.</h2>

<h3>Description</h3>

<p>Title: Filter out SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">FilterSNP(DataDir, ResultDir, finput, foutput, SNPvec, extract = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterSNP_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterSNP_:_ResultDir">ResultDir</code></td>
<td>
<p>A character string for the file path where all output files will be stored. The default is tempdir().</p>
</td></tr>
<tr><td><code id="FilterSNP_:_finput">finput</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="FilterSNP_:_foutput">foutput</code></td>
<td>
<p>Character string, specifying the prefix of the output PLINK binary files if the filtering option for the SNPs is chosen. The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="FilterSNP_:_SNPvec">SNPvec</code></td>
<td>
<p>R dataframe with SNP names to be excluded.</p>
</td></tr>
<tr><td><code id="FilterSNP_:_extract">extract</code></td>
<td>
<p>Boolean value, TRUE or FALSE, specifying whether to extract the snps or discard the snps. The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL. The filtered file will be saved in ResultDir.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
DataDir &lt;- system.file(&quot;extdata&quot;, package = &quot;GXwasR&quot;)
ResultDir &lt;- tempdir()
SNPvec &lt;- c(&quot;rs6529954&quot;, &quot;rs12858640&quot;, &quot;rs5962098&quot;)
finput &lt;- &quot;GXwasR_example&quot;
foutput &lt;- &quot;Filter_Test&quot;
FilterSNP(DataDir, ResultDir, finput, foutput, SNPvec = SNPvec,extract = TRUE)

</code></pre>


</div>
</div>
</div>
<div id="function-28" class="section level1">
<h1>Function 28</h1>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="fu">help</span>(DummyCovar, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>DummyCovar</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="DummyCovar">Function to recode a categorical variable to a set of binary dummy variables.</h2>

<h3>Description</h3>

<p>When dealing with categorical variables in genetic analysis using, a common approach is to convert these into dummy variables for proper analysis (1). This function creates K-1 new dummy variables for a variable with K categories. One level is automatically excluded from the dummy variables which serves as the reference category for subsequent analyses. This setup implicitly sets the excluded that category as the baseline against which other categories are compared.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DummyCovar(DataDir, bfile, incovar, outcovar)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DummyCovar_:_DataDir">DataDir</code></td>
<td>
<p>A character string for the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="DummyCovar_:_bfile">bfile</code></td>
<td>
<p>Character string, specifying the prefix of the input PLINK binary files for which covariate file will be generated.</p>
</td></tr>
<tr><td><code id="DummyCovar_:_incovar">incovar</code></td>
<td>
<p>Character string, specifying the prefix of the input covariate file. First two columns will be, FID (i.e., Family ID) and IID (i.e., Sample ID) and rest of the columns are covariates.</p>
</td></tr>
<tr><td><code id="DummyCovar_:_outcovar">outcovar</code></td>
<td>
<p>Character string, specifying the prefix of the Output covariate file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>R dataframe object with covariates
</p>


<h3>References</h3>

<p>(1) Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. The American Journal of Human Genetics, 81(3), 559-575. https://doi.org/10.1086/519795.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># bfile &lt;- Example_GXwasR
# incovar &lt;- &quot;covarfile_w_pc_age.txt&quot;
# outcovar &lt;- &quot;dummycovarfile&quot;

</code></pre>


</div>
</div>
</div>
<div id="function-29" class="section level1">
<h1>Function 29</h1>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="fu">help</span>(GXWASmiami, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>GXWASmiami</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="GXWASmiami">Miami plot</h2>

<h3>Description</h3>

<p>This function generates Miami plots for GWAS and XWAS
</p>


<h3>Usage</h3>

<pre><code class="language-R">GXWASmiami(
  ResultDir = tempdir(),
  FemaleWAS,
  MaleWAS,
  snp_pval = 1e-08,
  Xchr = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GXWASmiami_:_ResultDir">ResultDir</code></td>
<td>
<p>Character string for the folder path where the outputs will be saved.</p>
</td></tr>
<tr><td><code id="GXWASmiami_:_FemaleWAS">FemaleWAS</code></td>
<td>
<p>R dataframe of summary statistics of GWAS or XWAS of female samples with four columns, SNP(Variant),CHR(Chromosome number)
,POS(Base pair position) and pvalue(P-value of the test). This can be generated by running FM01comb or
FM02comb model with GXWAS function.</p>
</td></tr>
<tr><td><code id="GXWASmiami_:_MaleWAS">MaleWAS</code></td>
<td>
<p>R dataframe of summary statistics of GWAS or XWAS of male samples with four columns, SNP(Variant),CHR(Chromosome number)
,POS(Base pair position) and pvalue(P-value of the test). This can be generated by running FM01comb or
FM02comb model with GXWAS function.</p>
</td></tr>
<tr><td><code id="GXWASmiami_:_snp_pval">snp_pval</code></td>
<td>
<p>Numeric value as p-value threshold for annotation. SNPs below this p-value will be annotated on the plot. The default is 1e-08.</p>
</td></tr>
<tr><td><code id="GXWASmiami_:_Xchr">Xchr</code></td>
<td>
<p>Boolean value, TRUE or FALSE, specifying whether to generate Miami plot for stratified XWAS or not. The default is TRUE.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
</code></pre>


</div>
</div>
</div>
<div id="function-30" class="section level1">
<h1>Function 30</h1>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="fu">help</span>(Download_reference, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>Download_reference</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="Download_reference">Download Hapmap phase 3 and 1000 Genome phase 3 Data</h2>

<h3>Description</h3>

<p>Downloads reference data sets from specified URLs based on the reference dataset name and working directory.
Currently supports &#39;HapMapIII_NCBI36&#39; and &#39;ThousandGenome&#39;.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Download_reference(refdata, wdir = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Download_reference_:_refdata">refdata</code></td>
<td>
<p>A character string specifying the reference dataset to download. Should be one of &#39;HapMapIII_NCBI36&#39; or &#39;ThousandGenome&#39;.</p>
</td></tr>
<tr><td><code id="Download_reference_:_wdir">wdir</code></td>
<td>
<p>A character string specifying the working directory where the reference data will be downloaded and extracted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisible. The function prints a message upon successful download and extraction of the reference data.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Not run
#Download_reference(&quot;ThousandGenome&quot;, &quot;path/to/your/directory&quot;)
#Download_reference(&quot;HapMapIII_NCBI36&quot;, tempdir())

</code></pre>


</div>
</div>
</div>
<div id="function-31" class="section level1">
<h1>Function 31</h1>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">help</span>(SumstatGenCorr, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>SumstatGenCorr</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="SumstatGenCorr">SumstatGenCorr: Genetic Correlation Calculation from GWAS Summary Statistics</h2>

<h3>Description</h3>

<p>This function calculates the genetic correlation between two summary statistics
using a specified reference linkage disequilibrium (LD) matrix from the UK Biobank.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SumstatGenCorr(
  ResultDir = tempdir(),
  referenceLD,
  sumstat1,
  sumstat2,
  Nref = 335265,
  N0 = min(sumstat1$N),
  eigen.cut = &quot;automatic&quot;,
  lim = exp(-18),
  parallel = FALSE,
  numCores = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SumstatGenCorr_:_ResultDir">ResultDir</code></td>
<td>
<p>Directory where results should be saved.</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_referenceLD">referenceLD</code></td>
<td>
<p>Reference LD matrix identifier.
#&#39; These are the LD matrices and their eigen-decomposition from 335,265 genomic British UK Biobank individuals. Two sets of reference panel are provided:
</p>

<ol>
<li><p> 307,519 QCed UK Biobank Axiom Array SNPs. The size is about 7.5 GB after unzipping.
</p>
</li>
<li><p> 1,029,876 QCed UK Biobank imputed SNPs. The size is about 31 GB after unzipping. Although it takes more time, using the imputed panel provides more accurate estimates of genetic correlations.
Therefore if the GWAS includes most of the HapMap3 SNPs, then it is recommend using the imputed reference panel.
</p>
</li></ol>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_sumstat1">sumstat1</code></td>
<td>
<p>Data frame for the first set of summary statistics.
The input data frame should include following columns: SNP, SNP ID; A1, effect allele; A2, reference allele;
N, sample size; Z, z-score; If Z is not given, alternatively, you may provide: b, estimate of marginal effect in GWAS; se, standard error of the estimates of marginal effects in GWAS.</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_sumstat2">sumstat2</code></td>
<td>
<p>Data frame for the second set of summary statistics.
The input data frame should include following columns: SNP, SNP ID; A1, effect allele; A2, reference allele;
N, sample size; Z, z-score; If Z is not given, alternatively, you may provide: b, estimate of marginal effect in GWAS; se, standard error of the estimates of marginal effects in GWAS.</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_Nref">Nref</code></td>
<td>
<p>Sample size of the reference sample where LD is computed. If the default UK Biobank reference sample is used, Nref = 335265</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_N0">N0</code></td>
<td>
<p>Number of individuals included in both cohorts. The estimated genetic correlation is usually robust against misspecified N0.
If not given, the default value is set to the minimum sample size across all SNPs in cohort 1 and cohort 2.</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_eigen.cut">eigen.cut</code></td>
<td>
<p>Which eigenvalues and eigenvectors in each LD score matrix should be used for HDL.
Users are allowed to specify a numeric value between 0 and 1 for eigen.cut. For example, eigen.cut = 0.99 means using the leading eigenvalues explaining 99% of the variance
and their correspondent eigenvectors. If the default &#39;automatic&#39; is used, the eigen.cut gives the most stable heritability estimates will be used.</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_lim">lim</code></td>
<td>
<p>Tolerance limitation, default lim = exp(-18).</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_parallel">parallel</code></td>
<td>
<p>Boolean value, TRUE or FALSE for whether to perform parallel computation. The default is FALSE</p>
</td></tr>
<tr><td><code id="SumstatGenCorr_:_numCores">numCores</code></td>
<td>
<p>The number of cores to be used. The default is 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list is returned with:
</p>

<ul>
<li><p>rg The estimated genetic correlation.
</p>
</li>
<li><p>rg.se The standard error of the estimated genetic correlation.
</p>
</li>
<li><p>P P-value based on Wald test.
</p>
</li>
<li><p>estimates.df A detailed matrix includes the estimates and standard errors of heritabilities, genetic covariance and genetic correlation.
</p>
</li>
<li><p>eigen.use The eigen.cut used in computation.
</p>
</li></ul>



<h3>References</h3>

<p>Ning Z, Pawitan Y, Shen X (2020). High-definition likelihood inference of genetic correlations
across human complex traits.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Not Run
#ResultDir = tempdir()
#res &lt;- SumstatGenCorr(ResultDir, &quot;UKB_imputed_hapmap2_SVD_eigen99_extraction&quot;,
#                      &quot;sumstat1&quot;, &quot;sumstat2&quot;)
</code></pre>


</div>
</div>
</div>
<div id="function-32" class="section level1">
<h1>Function 32</h1>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">help</span>(LDPrune, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>LDPrune</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="LDPrune">LDPrune: Performs LD pruning on SNP data using PLINK</h2>

<h3>Description</h3>

<p>This function utilizes PLINK to perform LD pruning on genetic data. It identifies and removes SNPs that are in high linkage disequilibrium with each other within specified windows.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LDPrune(
  DataDir,
  finput,
  ResultDir = tempdir(),
  window_size = 50,
  step_size = 5,
  r2_threshold = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDPrune_:_DataDir">DataDir</code></td>
<td>
<p>Character string representing the file path of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="LDPrune_:_finput">finput</code></td>
<td>
<p>Character string specifying the prefix of the input PLINK binary files.</p>
</td></tr>
<tr><td><code id="LDPrune_:_ResultDir">ResultDir</code></td>
<td>
<p>Character string for the file path where all output files will be stored, defaulting to a temporary directory.</p>
</td></tr>
<tr><td><code id="LDPrune_:_window_size">window_size</code></td>
<td>
<p>Integer, specifying the number of SNPs to include in the sliding window.</p>
</td></tr>
<tr><td><code id="LDPrune_:_step_size">step_size</code></td>
<td>
<p>Integer, specifying the number of SNPs the window moves over in each step.</p>
</td></tr>
<tr><td><code id="LDPrune_:_r2_threshold">r2_threshold</code></td>
<td>
<p>Numeric, specifying the R^2 threshold for LD pruning.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector of SNP identifiers that remain after LD pruning or NULL if an error occurs.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Not Run
# DataDir &lt;- &quot;path/to/data&quot;
# ResultDir &lt;- tempdir()
# finput &lt;- &quot;dataset_prefix&quot;
# prunedSNPs &lt;- LDPrune(DataDir, finput, ResultDir, 50, 5, 0.2)
</code></pre>


</div>
</div>
</div>
<div id="function-33" class="section level1">
<h1>Function 33</h1>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="fu">help</span>(executePlinkMAF, <span class="at">package =</span> <span class="st">&quot;GXwasR&quot;</span>)</span></code></pre></div>
<div class="r-help-page"><div class="container">

<table style="width: 100%;"><tr><td>executePlinkMAF</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id="executePlinkMAF">Execute PLINK to Calculate Minor Allele Frequencies (MAF)</h2>

<h3>Description</h3>

<p>This function executes PLINK to calculate minor allele frequencies (MAF) for a given dataset. It sets up the necessary PLINK environment,
runs the PLINK command, and returns the MAF results as a DataFrame. Intermediate files generated by PLINK are cleaned up after execution.
</p>


<h3>Usage</h3>

<pre><code class="language-R">executePlinkMAF(DataDir, ResultDir, finput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="executePlinkMAF_:_DataDir">DataDir</code></td>
<td>
<p>Character. Directory containing the input PLINK files (.bed, .bim, .fam).</p>
</td></tr>
<tr><td><code id="executePlinkMAF_:_ResultDir">ResultDir</code></td>
<td>
<p>Character. Directory to store the output files generated by PLINK.</p>
</td></tr>
<tr><td><code id="executePlinkMAF_:_finput">finput</code></td>
<td>
<p>Character. Base name of the PLINK input files (without extensions).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame containing the minor allele frequency (MAF) results.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#don&#39;t run
#DataDir &lt;- &quot;path/to/data&quot;
#ResultDir &lt;- &quot;path/to/results&quot;
#finput &lt;- &quot;input_file_base_name&quot;
#maf_data &lt;- executePlinkMAF(DataDir, ResultDir, finput)
</code></pre>


</div>
</div>
</div>
<div id="tutorials" class="section level1">
<h1>Tutorials</h1>
<p>Please follow these tutorials to know more about the functionality of
the package GXwasR.</p>
<p>Tutorial for performing post-imputation QC followed by sex-aware
association tests: (vignette:Use of GXwasR)</p>
<p>Tutorial for performing pre-imputation QC using GXwasR:</p>
<p>Tutorial for performing post-imputation QC using GXwasR:</p>
<p>Tutorial for sex-combined and sex-stratified GWAS with
XWAS:<strong>hdhghchjfvjh</strong></p>
<p>Tutorial for sex-differential test:<strong>gdhgdcmh</strong></p>
<p>Tutorial for ancestry estimation:</p>
<p>Tutorial for computing polygenic risk
score:<strong>jcjjcjh</strong></p>
<p>Tutorial for meta analysis:</p>
<p>Tutorial for heritability estimate and genetic correlation:</p>
<p>Tutorial for gene-based tests:</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
