---
title: "GXwasR Functions"
author: "Banabithi Bose"
date: "2024-01-03"
output: 
  html_document:
  standalone: true
highlight: tango
self-contained: true
pdf_document:
  highlight: tango
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Functions Overview

This document lists all the functions in GXwasR package.It offers twenty-eight distinct functions, which are organized into six main categories: A) Pre-imputation QC, B) Post-imputation QC, C) Sex-combined and sex-stratified GWAS with specialized analysis for XWAS, D) Sex-differential test, E) High level analysis and a range of F) Utility Functions. These categories and their respective functions are detailed in the below Table.

## Table of Functions
| Function        | Description                                                                                                     | Category                             |
|-----------------|-----------------------------------------------------------------------------------------------------------------|--------------------------------------|
| QCsnp()         | Performs quality control for SNPs in PLINK binary files.                                                        | Pre-imputation QC, Post-imputation QC |
| QCsample()      | Identifies outlier individuals based on heterozygosity and missing genotype rates.                              | Pre-imputation QC, Post-imputation QC |
| AncestryCheck() | Evaluates samples' ancestry and flags outliers using PCA.                                                       | Pre-imputation QC                     |
| SexCheck()      | Compares sex assignments with predictions from X chromosome inbreeding coefficients.                            | Pre-imputation QC                     |
| Xhwe()          | Filters X-chromosome variants violating Hardy-Weinberg Equilibrium in females.                                  | Post-imputation QC                    |
| MAFdiffSexControl() | Tests for significant MAF differences between sexes in control samples.                                     | Post-imputation QC                    |
| FilterRegion()  | Filters out specific chromosomal regions from input PLINK files.                                                | Post-imputation QC, Utility Functions |
| GXwas()         | Runs GWAS models in autosomes with XWAS models like "FMcomb01", "FMcomb02", "FMstratified" for binary and quantitative traits, and "GWAScxci" for binary traits. Focuses on additive SNP effects, multi-collinearity issues, and includes multiple covariates and their interactions. | Sex-combined and sex-stratified GWAS with XWAS |
| PvalComb()      | Combines p-values from separate GWAS using various statistical methods and corrects summary p-values.           | Sex-combined and sex-stratified GWAS with XWAS |
| SexDiff()       | Evaluates sex differences in genetic effect size for each SNP.                                                  | Sex-differential test                 |
| SexDiffZscore() | Presumably analyzes sex differences using Z-score methodology, comparing genetic effect sizes between males and females. | Sex-differential test                 |
| DiffZeroOne()   | Assesses Z-scores for deviation from one and zero for statistics like genetic correlation.                      | Sex-differential test                 |
| TestXGene()     | Performs gene-based association tests using GWAS/XWAS summary statistics.                                       | High level analysis                   |
| MetaGXwas()     | Combines summary-level GWAS results using fixed-effect and random-effect models.                                | High level analysis                   |
| ComputePRS()    | Calculates polygenic risk scores from GWAS summary statistics.                                                  | High level analysis                   |
| ComputeCorrBT() | Likely computes correlation or association in a specific context related to binary traits.                      | High level analysis                   |
| EstimateHerit() | Computes SNP heritability using GREML or LDSC models.                                                          | High level analysis                   |
| SexRegress()    | Not previously described. Presumably involves regression analyses specific to sex-stratified data.             | High level analysis                   |
| FilterPlinkSample() | Prepares PLINK binary files with desired samples based on specified criteria.                                 | Utility Functions                     |
| ComputeGeneticPC() | Computes principal components from a genetic relationship matrix for population stratification correction.    | Utility Functions                     |
| ClumpLD()       | Performs linkage disequilibrium clumping of SNPs.                                                              | Utility Functions                     |
| GetMFPlink()    | Prepares separate male and female PLINK binary files from combined files.                                      | Utility Functions                     |
| plinkVCF()      | Converts VCF files to plink binary formats and vice versa, including creation of dummy FAM files.              | Utility Functions                     |
| MergeRegion()   | Combines two genotype datasets based on common or all SNPs.                                                    | Utility Functions                     |
| FilterAllele()  | Filters out multi-allelic variants from the genetic dataset, essential for maintaining dataset integrity and simplifying genetic analyses. | Utility Functions                     |
| PlinkSummary()  | Provides a summary of genotype datasets in plink format.                                                       | Utility Functions                     |
| FilterSNP()     | Filters out specific SNPs from the dataset based on user-defined criteria. | Utility Functions                     |
| DummyCovar()    | Recodes a categorical covariate into binary dummy variables for statistical analysis in GXwasR. | Utility Functions                     |
```{r}
library(GXwasR)
library(printr)
library(rmarkdown)
```

# Function 1
```{r}
help(QCsnp, package = "GXwasR")
```

# Function 2
```{r}
help(QCsample, package = "GXwasR")
```

# Function 3
```{r}
help(AncestryCheck, package = "GXwasR")
```

# Function 4
```{r}
help(SexCheck, package = "GXwasR")
```

# Function 5
```{r}
help(Xhwe, package = "GXwasR")
```

# Function 6
```{r}
help(MAFdiffSexControl, package = "GXwasR")
```

# Function 7
```{r}
help(FilterRegion, package = "GXwasR")
```

# Function 8
```{r}
help(GXwas, package = "GXwasR")
```

# Function 9
```{r}
help(PvalComb, package = "GXwasR")
```

# Function 10
```{r}
help(SexDiff, package = "GXwasR")
```

# Function 11
```{r}
help(SexDiffZscore, package = "GXwasR")
```

# Function 12
```{r}
help(DiffZeroOne, package = "GXwasR")
```

# Function 13
```{r}
help(TestXGene, package = "GXwasR")
```

# Function 14
```{r}
help(MetaGXwas, package = "GXwasR")
```

# Function 15
```{r}
help(ComputePRS, package = "GXwasR")
```

# Function 16
```{r}
help(ComputeCorrBT, package = "GXwasR")
```

# Function 17
```{r}
help(EstimateHerit, package = "GXwasR")
```

# Function 18
```{r}
help(SexRegress, package = "GXwasR")
```

# Function 19
```{r}
help(FilterPlinkSample, package = "GXwasR")
```

# Function 20
```{r}
help(ComputeGeneticPC, package = "GXwasR")
```

# Function 21
```{r}
help(ClumpLD, package = "GXwasR")
```

# Function 22
```{r}
help(GetMFPlink, package = "GXwasR")
```

# Function 23
```{r}
help(plinkVCF, package = "GXwasR")
```

# Function 24
```{r}
help(MergeRegion, package = "GXwasR")
```

# Function 25
```{r}
help(FilterAllele, package = "GXwasR")
```

# Function 26
```{r}
help(PlinkSummary, package = "GXwasR")
```

# Function 27
```{r}
help(FilterSNP, package = "GXwasR")
```

# Function 28
```{r}
help(DummyCovar, package = "GXwasR")
```
