---
title: "GWAS Meta analysis Using GXwasR"
author: "By Banabithi Bose, Northwestern University"
date: "1/13/2024"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{GWAS Meta analysis Using GXwasR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=8, fig.height=6)
```

## Meta analysis using GXwasR

Here we will present a tutorial showing how to perform meta analysis utilizing the R package GXwasR.

The example data sets for performing the tutorial can be accssed from "data" folder of the package.

## Example Datasets
Example datasets are provided with the package and can be accessed by calling data("GXwasRData").

Load the example datasets to perform this tutorial.
```{r}
## Load library
library(GXwasR)
data("GXwasRData")
```
(1) Two sets of summary GWAS summary statistics in .Rda files will be required to perform this tutorial. 
  **Summary_Stat_Ex1.Rda**
  **Summary_Stat_Ex2.Rda**
  

Check one of the summary statistics data
```{r}
## Visualize three rows and all the columns
Summary_Stat_Ex1[1:3,]
```
Among these 12 columns, some coulmns are mandatory for this tutorial, such as: ‘SNP’ (i.e., SNP idenitifier), ‘BETA’ (i.e., effect-size or logarithm of odds ratio), ‘SE’ (i.e., standard error of BETA), ‘P’ (i.e., p-values) and 'NMISS' (i.e., effective sample size). The other columns are, 'CHR' (i.e., chromosome code), 'BP' (i.e., base pair position), A1 (i.e., disease allele), TEST (i.e., association test type), L95 (i.e., lower limit of 95 percentile confidence interval), U95 (i.e., upper limit of 95 percentile confidence interval) and STAT (i.e., test statistic).

(2) **UniqueLoci.Rda:** .Rda file with a single column containing SNP names. These could be LD clumped SNPs or any other list of chosen SNPs for Meta analysis.

(3) **SNPsPlot.Rda:** .Rda file with a single column containing SNP names for the forest plots.

```{r}
## Call some other libraries for this vignette
library(printr)
library(rmarkdown)
```

## The function for meta analysis:
**MetaGWAS()**
```{r}
help(MetaGWAS,package = GXwasR)
```
## Running MetaGWAS
```{r}
DataDir = system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
SummData <- list(Summary_Stat_Ex1,Summary_Stat_Ex2)
SNPfile = "UniqueLoci"
useSNPposition = FALSE
UseA1 = TRUE
GCse = TRUE
byCHR = FALSE
pval_filter = "R"
top_snp_pval = 1e-08
max_top_snps = 10
chosen_snps_file = NULL
pval_threshold_manplot = 1e-05
plotname = "Meta_Analysis.plot"

x <- MetaGWAS(DataDir = DataDir, SummData = SummData,ResultDir=ResultDir, SNPfile = NULL, useSNPposition = TRUE, UseA1 = UseA1,GCse = GCse, plotname = "Meta_Analysis.plot", pval_filter, top_snp_pval, max_top_snps, chosen_snps_file = NULL, byCHR, pval_threshold_manplot)

```
## Random effect result
```{r}
# Dataframe with the fixed effect result
x1 <- x$Resultrandom
x2 <- x1[order(x1$P),]
knitr::kable(x2[1:10,], caption = 'Top ten associations from random effect model.')
```
## Fixed effect result
```{r}
# Dataframe with the fixed effect result
x1 <- x$Resultfixed
x2 <- x1[order(x1$P),]
knitr::kable(x2[1:10,], caption = 'Top ten associations from fixed effect model.')
```
## Weighted effect result
```{r}
# Dataframe with the fixed effect result
x1 <- x$Resultweighted
x2 <- x1[order(x1$P),]
knitr::kable(x2[1:10,], caption = 'Top ten associations from weighted effect model.')
```
## Metadata of the meta analysis
```{r}
# Dataframe with the metadata
x1 <- x$Metadata
x2 <- x1[order(x1$Q),]
knitr::kable(x2[1:10,], caption = 'Metadata of the top ten associations based on Cochrane’s Q statistics.')
```
## Problematic SNPs
```{r}
# Dataframe with the problematic SNPs
x1 <- x$ProblemSNP
knitr::kable(x1[1:10,], caption = 'Top ten probematic SNPs.')
```

## Including Plots

### Manhattan Plots and QQ plots:

```{r, echo=FALSE, out.width="1000px", out.height="500px"}
knitr::include_graphics(paste0(ResultDir,"/","Meta_Analysis.plot.jpeg"))
```

### All Forest plots
Now, let's see all the Forest plots.

```{r,echo=FALSE, out.height="500px",out.width="500px"}
knitr::include_graphics(paste0(ResultDir,"/","Meta_Analysis.plot.pdf"))
```
## Citation
If you are using GXwasR package and/or following this pipeline in your study, please cite it as:
```{r}
citation("GXwasR")
```

## Tutorials Link

**Please follow these tutorials to know more about the functionality of the package GXwasR.**

A list of all the functions in GXwasR:

Tutorial for performing post-imputation QC followed by sex-aware association tests:

Tutorial for performing pre-imputation QC using GXwasR:

Tutorial for performing post-imputation QC using GXwasR:

Tutorial for sex-combined and sex-stratified GWAS with XWAS:**hdhghchjfvjh**

Tutorial for sex-differential test:**gdhgdcmh**

Tutorial for ancestry estimation:

Tutorial for computing polygenic risk score:**jcjjcjh**

Tutorial for meta analysis:

Tutorial for heritability estimate and genetic correlation:

Tutorial for gene-based tests:

                                  ******************* THANK YOU! *************************




